<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Benchmarking phyloregion • phyloregion</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Benchmarking phyloregion">
<meta property="og:description" content="phyloregion">
<meta property="og:image" content="http://phyloregion.com/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phyloregion</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/phyloregion-intro.html">Vig 1: An introduction to the phyloregion package</a>
    </li>
    <li>
      <a href="../articles/Benchmark.html">Vig 2: Benchmarking phyloregion</a>
    </li>
    <li>
      <a href="../articles/ed.html">Vig 3: Fast Evolutionary Distinctiveness</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/darunabas/phyloregion/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Benchmarking phyloregion</h1>
                        <h4 data-toc-skip class="author">Barnabas H.
Daru, Piyal Karunarathne &amp; Klaus Schliep</h4>
            
            <h4 data-toc-skip class="date">November 15, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/darunabas/phyloregion/blob/HEAD/vignettes/Benchmark.Rmd" class="external-link"><code>vignettes/Benchmark.Rmd</code></a></small>
      <div class="hidden name"><code>Benchmark.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="benchmarking-phyloregion-with-comparable-packages">Benchmarking <code>phyloregion</code> with comparable packages<a class="anchor" aria-label="anchor" href="#benchmarking-phyloregion-with-comparable-packages"></a>
</h2>
<p>In this vignette, we benchmark <code>phyloregion</code> against other
similar <code>R</code> packages in analyses of standard alpha diversity
metrics commonly used in conservation, such as phylogenetic diversity
and phylogenetic endemism as well as metrics for analyzing compositional
turnover (e.g., beta diversity and phylogenetic beta diversity).
Specifically, we compare <code>phyloregion</code>’s functions with
available packages for efficiency in memory allocation and computation
speed in various biogeographic analyses.</p>
<p>First, load the packages for the benchmarking:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/" class="external-link">ape</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://Matrix.R-forge.R-project.org" class="external-link">Matrix</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bench.r-lib.org/" class="external-link">bench</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="co"># packages we benchmark</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/darunabas/phyloregion" class="external-link">phyloregion</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">betapart</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">picante</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegandevs/vegan" class="external-link">vegan</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/anttonalberdi/hilldiv" class="external-link">hilldiv</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BAT</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pez</span><span class="op">)</span></span></code></pre></div>
<p>We will use a small data set which comes with
<code>phyloregion</code>. This dataset consists of a dated phylogeny of
the woody plant species of southern Africa along with their geographical
distributions. The dataset comes from a study that maps tree diversity
hotspots in southern Africa <span class="citation">(Daru, Bank, and
Davies 2015)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">africa</span><span class="op">)</span></span>
<span><span class="co"># subset matrix</span></span>
<span><span class="va">X_sparse</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">comm</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, <span class="op">]</span></span>
<span><span class="va">X_sparse</span> <span class="op">&lt;-</span> <span class="va">X_sparse</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">]</span></span>
<span><span class="va">X_dense</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/coerce.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span></span>
<span><span class="va">Xt_dense</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/transpose.html" class="external-link">t</a></span><span class="op">(</span><span class="va">X_dense</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/object.size.html" class="external-link">object.size</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 76504 bytes</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/object.size.html" class="external-link">object.size</a></span><span class="op">(</span><span class="va">X_dense</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 134752 bytes</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  30 401</span></span></code></pre>
<p>To make results comparable, it is often desirable to make sure that
the taxa in different datasets match each other <span class="citation">(Kembel et al. 2010)</span>. For example, the community
matrix in the <code>hilldiv</code> package <span class="citation">(Alberdi 2019)</span> needs to be transposed. These
transformations can influence the execution times of the function, often
only marginally. Thus, to benchmark <code>phyloregion</code> against
other packages, we here use the package <code>bench</code> <span class="citation">(Hester 2020)</span> because it returns execution times
and provides estimates of memory allocations for each computation.</p>
</div>
<div class="section level2">
<h2 id="analysis-of-alpha-diversity">1. Analysis of alpha diversity<a class="anchor" aria-label="anchor" href="#analysis-of-alpha-diversity"></a>
</h2>
<div class="section level4">
<h4 id="benchmarking-phyloregion-for-analysis-of-phylogenetic-diversity">1.1. Benchmarking <code>phyloregion</code> for analysis of
phylogenetic diversity<a class="anchor" aria-label="anchor" href="#benchmarking-phyloregion-for-analysis-of-phylogenetic-diversity"></a>
</h4>
<p>For analysis of alpha diversity commonly used in conservation such as
phylogenetic diversity - the sum of all phylogenetic branch lengths
within an area <span class="citation">(Faith 1992)</span> -
<code>phyloregion</code> is 31 to 284 times faster and 67 to 192 times
memory efficient, compared to other packages!</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">phylo</span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/drop.tip.html" class="external-link">keep.tip</a></span><span class="op">(</span><span class="va">tree</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pd_picante</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">picante</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/picante/man/pd.html" class="external-link">pd</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>    <span class="va">res</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">pd_pez</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">pez</span><span class="fu">::</span><span class="fu">comparative.comm</span><span class="op">(</span><span class="va">tree</span>, <span class="va">x</span><span class="op">)</span></span>
<span>    <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">pez</span><span class="fu">::</span><span class="fu">.pd</span><span class="op">(</span><span class="va">dat</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>    <span class="va">res</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">pd_hilldiv</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">hilldiv</span><span class="fu">::</span><span class="fu">index_div</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span>, index<span class="op">=</span><span class="st">"faith"</span><span class="op">)</span></span>
<span><span class="va">pd_phyloregion</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">phyloregion</span><span class="fu">::</span><span class="fu"><a href="../reference/PD.html">PD</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res1</span> <span class="op">&lt;-</span> <span class="fu">bench</span><span class="fu">::</span><span class="fu">mark</span><span class="op">(</span>picante<span class="op">=</span><span class="fu">pd_picante</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,</span>
<span>          hilldiv<span class="op">=</span><span class="fu">pd_hilldiv</span><span class="op">(</span><span class="va">Xt_dense</span>,tree<span class="op">=</span><span class="va">tree</span><span class="op">)</span>,</span>
<span>          pez<span class="op">=</span><span class="fu">pd_pez</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,</span>
<span>          phyloregion<span class="op">=</span><span class="fu">pd_phyloregion</span><span class="op">(</span><span class="va">X_sparse</span>, <span class="va">tree</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## # A tibble: 4 × 6</span></span>
<span><span class="co">##   expression       min   median `itr/sec` mem_alloc `gc/sec`</span></span>
<span><span class="co">##   &lt;bch:expr&gt;  &lt;bch:tm&gt; &lt;bch:tm&gt;     &lt;dbl&gt; &lt;bch:byt&gt;    &lt;dbl&gt;</span></span>
<span><span class="co">## 1 picante      97.85ms 112.45ms      8.22    59.6MB     9.87</span></span>
<span><span class="co">## 2 hilldiv     762.42ms 762.42ms      1.31   170.1MB     3.93</span></span>
<span><span class="co">## 3 pez         108.59ms 122.83ms      8.16    60.4MB     8.16</span></span>
<span><span class="co">## 4 phyloregion   2.82ms   3.23ms    268.     909.8KB     5.99</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">res1</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="benchmark-phylo_diversity-1.png" alt="plot of chunk phylo_diversity"><div class="figcaption">plot of chunk phylo_diversity</div>
</div>
</div>
<div class="section level4">
<h4 id="benchmarking-phyloregion-for-analysis-of-phylogenetic-endemism">1.2. Benchmarking <code>phyloregion</code> for analysis of
phylogenetic endemism<a class="anchor" aria-label="anchor" href="#benchmarking-phyloregion-for-analysis-of-phylogenetic-endemism"></a>
</h4>
<p>Another benchmark for <code>phyloregion</code> is in the analysis of
phylogenetic endemism, the degree to which phylogenetic diversity is
restricted to any given area <span class="citation">(Rosauer et al.
2009)</span>. Here, we found that <code>phyloregion</code> is 160 times
faster and 489 times efficient in memory allocation.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">phylo</span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/drop.tip.html" class="external-link">keep.tip</a></span><span class="op">(</span><span class="va">tree</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pe_pez</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">pez</span><span class="fu">::</span><span class="fu">comparative.comm</span><span class="op">(</span><span class="va">tree</span>, <span class="va">x</span><span class="op">)</span></span>
<span>    <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">pez</span><span class="fu">::</span><span class="fu">pez.endemism</span><span class="op">(</span><span class="va">dat</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>    <span class="va">res</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">pe_phyloregion</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">phyloregion</span><span class="fu">::</span><span class="fu"><a href="../reference/phylo_endemism.html">phylo_endemism</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res2</span> <span class="op">&lt;-</span> <span class="fu">bench</span><span class="fu">::</span><span class="fu">mark</span><span class="op">(</span>pez<span class="op">=</span><span class="fu">pe_pez</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,</span>
<span>          phyloregion<span class="op">=</span><span class="fu">pe_phyloregion</span><span class="op">(</span><span class="va">X_sparse</span>, <span class="va">tree</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## # A tibble: 2 × 6</span></span>
<span><span class="co">##   expression       min   median `itr/sec` mem_alloc `gc/sec`</span></span>
<span><span class="co">##   &lt;bch:expr&gt;  &lt;bch:tm&gt; &lt;bch:tm&gt;     &lt;dbl&gt; &lt;bch:byt&gt;    &lt;dbl&gt;</span></span>
<span><span class="co">## 1 pez         630.38ms 630.38ms      1.59  493.88MB     9.52</span></span>
<span><span class="co">## 2 phyloregion   2.94ms   3.19ms    280.      1.06MB     3.98</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">res2</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="benchmark-phylo_endemism-1.png" alt="plot of chunk phylo_endemism"><div class="figcaption">plot of chunk phylo_endemism</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="analysis-of-compositional-turnover-beta-diversity">2. Analysis of compositional turnover (beta diversity)<a class="anchor" aria-label="anchor" href="#analysis-of-compositional-turnover-beta-diversity"></a>
</h2>
<div class="section level4">
<h4 id="benchmarking-phyloregion-for-analysis-of-taxonomic-beta-diversity">2.1. Benchmarking <code>phyloregion</code> for analysis of taxonomic
beta diversity<a class="anchor" aria-label="anchor" href="#benchmarking-phyloregion-for-analysis-of-taxonomic-beta-diversity"></a>
</h4>
<p>For analysis of taxonomic beta diversity, which compares diversity
between communities <span class="citation">(Koleff, Gaston, and Lennon
2003)</span>, <code>phyloregion</code> has marginal advantage over other
packages. Nonetheless, it is 1-39 times faster and allocates 2 to 110
times less memory than other packages.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chk_fun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">target</span>, <span class="va">current</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/all.equal-methods.html" class="external-link">all.equal</a></span><span class="op">(</span><span class="va">target</span>, <span class="va">current</span>, check.attributes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fun_phyloregion</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/coerce.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu">phyloregion</span><span class="fu">::</span><span class="fu"><a href="../reference/beta_diss.html">beta_diss</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">fun_betapart</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/coerce.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu">betapart</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/betapart/man/beta.pair.html" class="external-link">beta.pair</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">fun_vegan</span>  <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/coerce.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu">vegan</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></span><span class="op">(</span><span class="va">x</span>, binary<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fun_BAT</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/coerce.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu">BAT</span><span class="fu">::</span><span class="fu">beta</span><span class="op">(</span><span class="va">x</span>, func <span class="op">=</span> <span class="st">"Soerensen"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">res3</span> <span class="op">&lt;-</span> <span class="fu">bench</span><span class="fu">::</span><span class="fu">mark</span><span class="op">(</span>phyloregion<span class="op">=</span><span class="fu">fun_phyloregion</span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span>,</span>
<span>                    betapart<span class="op">=</span><span class="fu">fun_betapart</span><span class="op">(</span><span class="va">X_dense</span><span class="op">)</span>,</span>
<span>                    vegan<span class="op">=</span><span class="fu">fun_vegan</span><span class="op">(</span><span class="va">X_dense</span><span class="op">)</span>,</span>
<span>                    BAT<span class="op">=</span><span class="fu">fun_BAT</span><span class="op">(</span><span class="va">X_dense</span><span class="op">)</span>, check<span class="op">=</span><span class="va">chk_fun</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## # A tibble: 4 × 6</span></span>
<span><span class="co">##   expression       min   median `itr/sec` mem_alloc `gc/sec`</span></span>
<span><span class="co">##   &lt;bch:expr&gt;  &lt;bch:tm&gt; &lt;bch:tm&gt;     &lt;dbl&gt; &lt;bch:byt&gt;    &lt;dbl&gt;</span></span>
<span><span class="co">## 1 phyloregion  598.5µs    689µs    1210.    428.2KB     4.39</span></span>
<span><span class="co">## 2 betapart     849.4µs    909µs    1024.    594.1KB    10.3 </span></span>
<span><span class="co">## 3 vegan        946.8µs    992µs     958.   1016.1KB     7.01</span></span>
<span><span class="co">## 4 BAT           44.5ms     47ms      20.9    31.8MB     8.95</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">res3</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="benchmark-beta_diversity-1.png" alt="plot of chunk beta_diversity"><div class="figcaption">plot of chunk beta_diversity</div>
</div>
</div>
<div class="section level4">
<h4 id="benchmarking-phyloregion-for-analysis-of-phylogenetic-beta-diversity">2.2. Benchmarking <code>phyloregion</code> for analysis of
phylogenetic beta diversity<a class="anchor" aria-label="anchor" href="#benchmarking-phyloregion-for-analysis-of-phylogenetic-beta-diversity"></a>
</h4>
<p>For analysis of phylogenetic turnover (beta-diversity) among
communities - the proportion of shared phylogenetic branch lengths
between communities <span class="citation">(Graham and Fine 2008)</span>
- <code>phyloregion</code> is 300-400 times faster and allocates 100-600
times less memory!</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fun_phyloregion</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">phyloregion</span><span class="fu">::</span><span class="fu"><a href="../reference/phylobeta.html">phylobeta</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">fun_betapart</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">betapart</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/betapart/man/phylo.beta.pair.html" class="external-link">phylo.beta.pair</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">fun_picante</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fl">1</span> <span class="op">-</span> <span class="fu">picante</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/picante/man/phylosor.html" class="external-link">phylosor</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span></span>
<span><span class="va">fun_BAT</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">BAT</span><span class="fu">::</span><span class="fu">beta</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span>, func <span class="op">=</span> <span class="st">"Soerensen"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">chk_fun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">target</span>, <span class="va">current</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/all.equal-methods.html" class="external-link">all.equal</a></span><span class="op">(</span><span class="va">target</span>, <span class="va">current</span>, check.attributes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res4</span> <span class="op">&lt;-</span> <span class="fu">bench</span><span class="fu">::</span><span class="fu">mark</span><span class="op">(</span>picante<span class="op">=</span><span class="fu">fun_picante</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,</span>
<span>                       betapart<span class="op">=</span><span class="fu">fun_betapart</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,</span>
<span>                       BAT<span class="op">=</span><span class="fu">fun_BAT</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,</span>
<span>                       phyloregion<span class="op">=</span><span class="fu">fun_phyloregion</span><span class="op">(</span><span class="va">X_sparse</span>, <span class="va">tree</span><span class="op">)</span>, check<span class="op">=</span><span class="va">chk_fun</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## # A tibble: 4 × 6</span></span>
<span><span class="co">##   expression       min   median `itr/sec` mem_alloc `gc/sec`</span></span>
<span><span class="co">##   &lt;bch:expr&gt;  &lt;bch:tm&gt; &lt;bch:tm&gt;     &lt;dbl&gt; &lt;bch:byt&gt;    &lt;dbl&gt;</span></span>
<span><span class="co">## 1 picante        2.25s    2.25s     0.444    1.24GB    2.66 </span></span>
<span><span class="co">## 2 betapart       2.05s    2.05s     0.489    1.24GB    2.93 </span></span>
<span><span class="co">## 3 BAT            1.41s    1.41s     0.712  293.64MB    0.712</span></span>
<span><span class="co">## 4 phyloregion   4.21ms   4.49ms   175.       1.12MB    1.99</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">res4</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="benchmark-phylobeta-1.png" alt="plot of chunk phylobeta"><div class="figcaption">plot of chunk phylobeta</div>
</div>
<p>Note that for this test, <code>picante</code> returns a similarity
matrix while <code>betapart</code>, and <code>phyloregion</code> return
a dissimilarity matrix.</p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Monterey 12.6</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] pez_1.2-4         BAT_2.9.2         hilldiv_1.5.1     picante_1.8.2    </span></span>
<span><span class="co">##  [5] nlme_3.1-157      vegan_2.6-2       lattice_0.20-45   permute_0.9-7    </span></span>
<span><span class="co">##  [9] betapart_1.5.6    phyloregion_1.0.7 bench_1.1.2       Matrix_1.5-3     </span></span>
<span><span class="co">## [13] ape_5.6-2         knitr_1.39        ggplot2_3.3.6    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] utf8_1.2.2              proto_1.0.0             ks_1.13.5              </span></span>
<span><span class="co">##   [4] tidyselect_1.2.0        htmlwidgets_1.5.4       grid_4.2.1             </span></span>
<span><span class="co">##   [7] combinat_0.0-8          pROC_1.18.0             animation_2.7          </span></span>
<span><span class="co">##  [10] munsell_0.5.0           codetools_0.2-18        clustMixType_0.2-15    </span></span>
<span><span class="co">##  [13] interp_1.1-3            future_1.29.0           withr_2.5.0            </span></span>
<span><span class="co">##  [16] profmem_0.6.0           colorspace_2.0-3        highr_0.9              </span></span>
<span><span class="co">##  [19] rstudioapi_0.13         geometry_0.4.6.1        stats4_4.2.1           </span></span>
<span><span class="co">##  [22] ggsignif_0.6.4          listenv_0.8.0           slam_0.1-50            </span></span>
<span><span class="co">##  [25] FD_1.0-12.1             nls2_0.3-3              mnormt_2.1.0           </span></span>
<span><span class="co">##  [28] farver_2.1.1            coda_0.19-4             parallelly_1.32.1      </span></span>
<span><span class="co">##  [31] vctrs_0.5.0             generics_0.1.3          clusterGeneration_1.3.7</span></span>
<span><span class="co">##  [34] ipred_0.9-13            xfun_0.31               itertools_0.1-3        </span></span>
<span><span class="co">##  [37] fastcluster_1.2.3       R6_2.5.1                doParallel_1.0.17      </span></span>
<span><span class="co">##  [40] ggbeeswarm_0.6.0        pdist_1.2.1             assertthat_0.2.1       </span></span>
<span><span class="co">##  [43] scales_1.2.0            nnet_7.3-17             beeswarm_0.4.0         </span></span>
<span><span class="co">##  [46] rgeos_0.5-9             gtable_0.3.0            globals_0.16.1         </span></span>
<span><span class="co">##  [49] caper_1.0.1             phangorn_2.9.0          MatrixModels_0.5-1     </span></span>
<span><span class="co">##  [52] timeDate_4021.106       rlang_1.0.6             FSA_0.9.3              </span></span>
<span><span class="co">##  [55] scatterplot3d_0.3-41    splines_4.2.1           rstatix_0.7.1          </span></span>
<span><span class="co">##  [58] rgdal_1.5-30            ModelMetrics_1.2.2.2    broom_1.0.1            </span></span>
<span><span class="co">##  [61] checkmate_2.1.0         reshape2_1.4.4          abind_1.4-5            </span></span>
<span><span class="co">##  [64] backports_1.4.1         Hmisc_4.7-1             caret_6.0-93           </span></span>
<span><span class="co">##  [67] tools_4.2.1             lava_1.7.0              psych_2.2.9            </span></span>
<span><span class="co">##  [70] lavaan_0.6-12           ellipsis_0.3.2          raster_3.5-21          </span></span>
<span><span class="co">##  [73] RColorBrewer_1.1-3      proxy_0.4-27            Rcpp_1.0.9             </span></span>
<span><span class="co">##  [76] plyr_1.8.7              base64enc_0.1-3         progress_1.2.2         </span></span>
<span><span class="co">##  [79] purrr_0.3.4             prettyunits_1.1.1       ggpubr_0.4.0           </span></span>
<span><span class="co">##  [82] rpart_4.1.16            deldir_1.0-6            pbapply_1.5-0          </span></span>
<span><span class="co">##  [85] deSolve_1.34            qgraph_1.9.2            cluster_2.1.3          </span></span>
<span><span class="co">##  [88] magrittr_2.0.3          data.table_1.14.2       SparseM_1.81           </span></span>
<span><span class="co">##  [91] mvtnorm_1.1-3           smoothr_0.2.2           hms_1.1.1              </span></span>
<span><span class="co">##  [94] evaluate_0.15           jpeg_0.1-9              mclust_6.0.0           </span></span>
<span><span class="co">##  [97] gridExtra_2.3           compiler_4.2.1          tibble_3.1.8           </span></span>
<span><span class="co">## [100] maps_3.4.0              KernSmooth_2.23-20      crayon_1.5.1           </span></span>
<span><span class="co">## [103] hypervolume_3.0.4       htmltools_0.5.3         mgcv_1.8-40            </span></span>
<span><span class="co">## [106] corpcor_1.6.10          Formula_1.2-4           snow_0.4-4             </span></span>
<span><span class="co">## [109] tidyr_1.2.0             expm_0.999-6            lubridate_1.8.0        </span></span>
<span><span class="co">## [112] DBI_1.1.3               magic_1.6-0             subplex_1.8            </span></span>
<span><span class="co">## [115] MASS_7.3-57             ade4_1.7-19             car_3.1-1              </span></span>
<span><span class="co">## [118] cli_3.4.1               quadprog_1.5-8          parallel_4.2.1         </span></span>
<span><span class="co">## [121] gower_1.0.0             igraph_1.3.4            pkgconfig_2.0.3        </span></span>
<span><span class="co">## [124] numDeriv_2016.8-1.1     foreign_0.8-82          sp_1.5-0               </span></span>
<span><span class="co">## [127] terra_1.5-21            recipes_1.0.3           foreach_1.5.2          </span></span>
<span><span class="co">## [130] pbivnorm_0.6.0          predicts_0.1-3          vipor_0.4.5            </span></span>
<span><span class="co">## [133] hardhat_1.2.0           prodlim_2019.11.13      stringr_1.4.0          </span></span>
<span><span class="co">## [136] digest_0.6.29           pracma_2.4.2            phytools_1.0-3         </span></span>
<span><span class="co">## [139] rcdd_1.5                fastmatch_1.1-3         htmlTable_2.4.1        </span></span>
<span><span class="co">## [142] quantreg_5.94           gtools_3.9.3            geiger_2.0.10          </span></span>
<span><span class="co">## [145] lifecycle_1.0.3         glasso_1.11             carData_3.0-5          </span></span>
<span><span class="co">## [148] maptpx_1.9-7            fansi_1.0.3             pillar_1.8.0           </span></span>
<span><span class="co">## [151] fastmap_1.1.0           plotrix_3.8-2           survival_3.3-1         </span></span>
<span><span class="co">## [154] glue_1.6.2              fdrtool_1.2.17          png_0.1-7              </span></span>
<span><span class="co">## [157] iterators_1.0.14        class_7.3-20            stringi_1.7.8          </span></span>
<span><span class="co">## [160] palmerpenguins_0.1.1    doSNOW_1.0.20           latticeExtra_0.6-30    </span></span>
<span><span class="co">## [163] dplyr_1.0.9             e1071_1.7-11            future.apply_1.10.0</span></span></code></pre>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">REFERENCES<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-hilldiv" class="csl-entry">
Alberdi, Antton. 2019. <span>“Hilldiv: An r Package for the Integral
Analysis of Diversity Based on Hill Numbers.”</span> <em>bioRxiv</em>.
<a href="https://www.biorxiv.org/content/10.1101/545665v1" class="external-link">https://www.biorxiv.org/content/10.1101/545665v1</a>.
</div>
<div id="ref-Daru2015ddi" class="csl-entry">
Daru, Barnabas H., Michelle van der Bank, and T. Jonathan Davies. 2015.
<span>“Spatial Incongruence Among Hotspots and Complementary Areas of
Tree Diversity in Southern Africa.”</span> <em>Diversity and
Distributions</em> 21 (7): 769–80. <a href="https://doi.org/10.1111/ddi.12290" class="external-link">https://doi.org/10.1111/ddi.12290</a>.
</div>
<div id="ref-Faith1992" class="csl-entry">
Faith, Daniel P. 1992. <span>“Conservation Evaluation and Phylogenetic
Diversity.”</span> <em>Biological Conservation</em> 61 (1): 1–10. <a href="https://doi.org/10.1016/0006-3207(92)91201-3" class="external-link">https://doi.org/10.1016/0006-3207(92)91201-3</a>.
</div>
<div id="ref-Graham2008" class="csl-entry">
Graham, Catherine H., and Paul V. A. Fine. 2008. <span>“Phylogenetic
Beta Diversity: Linking Ecological and Evolutionary Processes Across
Space in Time.”</span> <em>Ecology Letters</em> 11 (12): 1265–77. <a href="https://doi.org/10.1111/j.1461-0248.2008.01256.x" class="external-link">https://doi.org/10.1111/j.1461-0248.2008.01256.x</a>.
</div>
<div id="ref-bench2020" class="csl-entry">
Hester, Jim. 2020. <em>Bench: High Precision Timing of r
Expressions</em>. <a href="https://CRAN.R-project.org/package=bench" class="external-link">https://CRAN.R-project.org/package=bench</a>.
</div>
<div id="ref-Kembel2010" class="csl-entry">
Kembel, S. W., P. D. Cowan, M. R. Helmus, W. K. Cornwell, H. Morlon, D.
D. Ackerly, S. P. Blomberg, and C. O. Webb. 2010. <span>“Picante:
<span>R</span> Tools for Integrating Phylogenies and Ecology.”</span>
<em>Bioinformatics</em> 26: 1463–64.
</div>
<div id="ref-Koleff2003" class="csl-entry">
Koleff, Patricia, Kevin J. Gaston, and Jack J. Lennon. 2003.
<span>“Measuring Beta Diversity for Presence–Absence Data.”</span>
<em>Journal of Animal Ecology</em> 72 (3): 367–82. <a href="https://doi.org/10.1046/j.1365-2656.2003.00710.x" class="external-link">https://doi.org/10.1046/j.1365-2656.2003.00710.x</a>.
</div>
<div id="ref-Rosauer2009" class="csl-entry">
Rosauer, Dan, Shawn W. Laffan, Michael D. Crisp, Stephen C. Donnellan,
and Lyn G. Cook. 2009. <span>“Phylogenetic Endemism: A New Approach for
Identifying Geographical Concentrations of Evolutionary History.”</span>
<em>Molecular Ecology</em> 18 (19): 4061–72. <a href="https://doi.org/10.1111/j.1365-294X.2009.04311.x" class="external-link">https://doi.org/10.1111/j.1365-294X.2009.04311.x</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://darulab.org/" class="external-link">Barnabas H. Daru</a>, <a href="https://github.com/piyalkarum/" class="external-link">Piyal Karunarathne</a>, <a href="https://kschliep.netlify.app/" class="external-link">Klaus Schliep</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
