<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Benchmarking phyloregion • phyloregion</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Benchmarking phyloregion">
<meta property="og:description" content="phyloregion">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phyloregion</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Benchmark.html">Benchmarking phyloregion</a>
    </li>
    <li>
      <a href="../articles/ed.html">Fast Evolutionary Distinctiveness in R</a>
    </li>
    <li>
      <a href="../articles/heterogeneity.html">Endemism patterns are scale dependent</a>
    </li>
    <li>
      <a href="../articles/pd.html">Phylogenetic Diversity for Large Datasets</a>
    </li>
    <li>
      <a href="../articles/phyloregion-intro.html">An introduction to the phyloregion package</a>
    </li>
    <li>
      <a href="../articles/scalability.html">Scalability of phyloregion</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/darunabas/phyloregion/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Benchmark_files/accessible-code-block-0.0.1/empty-anchor.js"></script><link href="Benchmark_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="Benchmark_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Benchmarking phyloregion</h1>
                        <h4 class="author">Barnabas H. Daru, Piyal Karunarathne &amp; Klaus Schliep</h4>
            
            <h4 class="date">July 19, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/darunabas/phyloregion/blob/master/vignettes/Benchmark.Rmd"><code>vignettes/Benchmark.Rmd</code></a></small>
      <div class="hidden name"><code>Benchmark.Rmd</code></div>

    </div>

    
    
<div id="benchmarking-phyloregion-with-comparable-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#benchmarking-phyloregion-with-comparable-packages" class="anchor"></a>Benchmarking <code>phyloregion</code> with comparable packages</h2>
<p>In this vignette, we benchmark <code>phyloregion</code> against other similar <code>R</code> packages in analyses of standard alpha diversity metrics commonly used in conservation, such as phylogenetic diversity and phylogenetic endemism as well as metrics for analyzing compositional turnover (e.g., beta diversity and phylogenetic beta diversity). Specifically, we compare <code>phyloregion</code>’s functions with available packages for efficiency in memory allocation and computation speed in various biogeographic analyses.</p>
<p>First, load the packages for the benchmarking:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/">ape</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://Matrix.R-forge.R-project.org/">Matrix</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/r-lib/bench">bench</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="co"># packages we benchmark</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/darunabas/phyloregion">phyloregion</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">betapart</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">picante</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://cran.r-project.org">vegan</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/anttonalberdi/hilldiv">hilldiv</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BAT</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">pez</span><span class="op">)</span></code></pre></div>
<p>We will use a small data set which comes with <code>phyloregion</code>. This dataset consists of a dated phylogeny of the woody plant species of southern Africa along with their geographical distributions. The dataset comes from a study that maps tree diversity hotspots in southern Africa.<span class="citation"><sup>1</sup></span></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">africa</span><span class="op">)</span>
<span class="co"># subset matrix</span>
<span class="va">X_sparse</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">comm</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, <span class="op">]</span>
<span class="va">X_sparse</span> <span class="op">&lt;-</span> <span class="va">X_sparse</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html">colSums</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">]</span>
<span class="va">X_dense</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/raster/man/as.matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span>
<span class="va">Xt_dense</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/raster/man/transpose.html">t</a></span><span class="op">(</span><span class="va">X_dense</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/object.size.html">object.size</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span></code></pre></div>
<pre><code>## 76504 bytes</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/object.size.html">object.size</a></span><span class="op">(</span><span class="va">X_dense</span><span class="op">)</span></code></pre></div>
<pre><code>## 134752 bytes</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/raster/man/dimensions.html">dim</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span></code></pre></div>
<pre><code>## [1]  30 401</code></pre>
<p>To make results comparable, it is often desirable to make sure that the taxa in different datasets match each other.<span class="citation"><sup>2</sup></span> For example, the community matrix in the <code>hilldiv</code> package<span class="citation"><sup>3</sup></span> needs to be transposed. These transformations can influence the execution times of the function, often only marginally. Thus, to benchmark <code>phyloregion</code> against other packages, we here use the package <code>bench</code><span class="citation"><sup>4</sup></span> because it returns execution times and provides estimates of memory allocations for each computation.</p>
</div>
<div id="analysis-of-alpha-diversity" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis-of-alpha-diversity" class="anchor"></a>1. Analysis of alpha diversity</h2>
<div id="benchmarking-phyloregion-for-analysis-of-phylogenetic-diversity" class="section level4">
<h4 class="hasAnchor">
<a href="#benchmarking-phyloregion-for-analysis-of-phylogenetic-diversity" class="anchor"></a>1.1. Benchmarking <code>phyloregion</code> for analysis of phylogenetic diversity</h4>
<p>For analysis of alpha diversity commonly used in conservation such as phylogenetic diversity - the sum of all phylogenetic branch lengths within an area<span class="citation"><sup>5</sup></span> - <code>phyloregion</code> is 31 to 284 times faster and 67 to 192 times memory efficient, compared to other packages!</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tree</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">phylo</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/drop.tip.html">keep.tip</a></span><span class="op">(</span><span class="va">tree</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span><span class="op">)</span>

<span class="va">pd_picante</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">{</span>
    <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">picante</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/picante/man/pd.html">pd</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>
    <span class="fu"><a href="https://rdrr.io/pkg/raster/man/names.html">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
    <span class="va">res</span>
<span class="op">}</span>

<span class="va">pd_pez</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">{</span>
    <span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">pez</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pez/man/comparative.comm.html">comparative.comm</a></span><span class="op">(</span><span class="va">tree</span>, <span class="va">x</span><span class="op">)</span>
    <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">pez</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pez/man/pez.metrics.html">.pd</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>
    <span class="fu"><a href="https://rdrr.io/pkg/raster/man/names.html">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
    <span class="va">res</span>
<span class="op">}</span>

<span class="va">pd_hilldiv</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">hilldiv</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/hilldiv/man/index_div.html">index_div</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span>, index<span class="op">=</span><span class="st">"faith"</span><span class="op">)</span>
<span class="va">pd_phyloregion</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">phyloregion</span><span class="fu">::</span><span class="fu"><a href="../reference/PD.html">PD</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span>

<span class="va">res1</span> <span class="op">&lt;-</span> <span class="fu">bench</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bench/man/mark.html">mark</a></span><span class="op">(</span>picante<span class="op">=</span><span class="fu">pd_picante</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,
          hilldiv<span class="op">=</span><span class="fu">pd_hilldiv</span><span class="op">(</span><span class="va">Xt_dense</span>,tree<span class="op">=</span><span class="va">tree</span><span class="op">)</span>,
          pez<span class="op">=</span><span class="fu">pd_pez</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,
          phyloregion<span class="op">=</span><span class="fu">pd_phyloregion</span><span class="op">(</span><span class="va">X_sparse</span>, <span class="va">tree</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">res1</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 4 x 6
##   expression       min   median `itr/sec` mem_alloc `gc/sec`
##   &lt;bch:expr&gt;  &lt;bch:tm&gt; &lt;bch:tm&gt;     &lt;dbl&gt; &lt;bch:byt&gt;    &lt;dbl&gt;
## 1 picante     118.88ms 125.72ms     7.34    59.62MB    12.8 
## 2 hilldiv         1.1s     1.1s     0.911  170.22MB     4.56
## 3 pez         125.34ms 139.49ms     7.27    60.89MB    14.5 
## 4 phyloregion   3.21ms   4.61ms   142.       1.86MB     4.00</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html">autoplot</a></span><span class="op">(</span><span class="va">res1</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="benchmark-phylo_diversity-1.png" alt=""><p class="caption">plot of chunk phylo_diversity</p>
</div>
</div>
<div id="benchmarking-phyloregion-for-analysis-of-phylogenetic-endemism" class="section level4">
<h4 class="hasAnchor">
<a href="#benchmarking-phyloregion-for-analysis-of-phylogenetic-endemism" class="anchor"></a>1.2. Benchmarking <code>phyloregion</code> for analysis of phylogenetic endemism</h4>
<p>Another benchmark for <code>phyloregion</code> is in the analysis of phylogenetic endemism, the degree to which phylogenetic diversity is restricted to any given area.<span class="citation"><sup>6</sup></span> Here, we found that <code>phyloregion</code> is 160 times faster and 489 times efficient in memory allocation.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tree</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">phylo</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/drop.tip.html">keep.tip</a></span><span class="op">(</span><span class="va">tree</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span><span class="op">)</span>

<span class="va">pe_pez</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">{</span>
    <span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">pez</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pez/man/comparative.comm.html">comparative.comm</a></span><span class="op">(</span><span class="va">tree</span>, <span class="va">x</span><span class="op">)</span>
    <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">pez</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pez/man/pez.endemism.html">pez.endemism</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>
    <span class="fu"><a href="https://rdrr.io/pkg/raster/man/names.html">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
    <span class="va">res</span>
<span class="op">}</span>

<span class="va">pe_phyloregion</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">phyloregion</span><span class="fu">::</span><span class="fu"><a href="../reference/phylo_endemism.html">phylo_endemism</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span>

<span class="va">res2</span> <span class="op">&lt;-</span> <span class="fu">bench</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bench/man/mark.html">mark</a></span><span class="op">(</span>pez<span class="op">=</span><span class="fu">pe_pez</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,
          phyloregion<span class="op">=</span><span class="fu">pe_phyloregion</span><span class="op">(</span><span class="va">X_sparse</span>, <span class="va">tree</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">res2</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 2 x 6
##   expression       min   median `itr/sec` mem_alloc `gc/sec`
##   &lt;bch:expr&gt;  &lt;bch:tm&gt; &lt;bch:tm&gt;     &lt;dbl&gt; &lt;bch:byt&gt;    &lt;dbl&gt;
## 1 pez            1.38s    1.38s     0.725  493.84MB     4.35
## 2 phyloregion   3.24ms   3.47ms   265.       1.07MB     1.99</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html">autoplot</a></span><span class="op">(</span><span class="va">res2</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="benchmark-phylo_endemism-1.png" alt=""><p class="caption">plot of chunk phylo_endemism</p>
</div>
</div>
</div>
<div id="analysis-of-compositional-turnover-beta-diversity" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis-of-compositional-turnover-beta-diversity" class="anchor"></a>2. Analysis of compositional turnover (beta diversity)</h2>
<div id="benchmarking-phyloregion-for-analysis-of-taxonomic-beta-diversity" class="section level4">
<h4 class="hasAnchor">
<a href="#benchmarking-phyloregion-for-analysis-of-taxonomic-beta-diversity" class="anchor"></a>2.1. Benchmarking <code>phyloregion</code> for analysis of taxonomic beta diversity</h4>
<p>For analysis of taxonomic beta diversity, which compares diversity between communities,<span class="citation"><sup>7</sup></span> <code>phyloregion</code> has marginal advantage over other packages. Nonetheless, it is 1-39 times faster and allocates 2 to 110 times less memory than other packages.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">chk_fun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">target</span>, <span class="va">current</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/all.equal.html">all.equal</a></span><span class="op">(</span><span class="va">target</span>, <span class="va">current</span>, check.attributes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="va">fun_phyloregion</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/raster/man/as.matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">phyloregion</span><span class="fu">::</span><span class="fu"><a href="../reference/beta_core.html">beta_diss</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="va">fun_betapart</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/raster/man/as.matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">betapart</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/betapart/man/beta.pair.html">beta.pair</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="va">fun_vegan</span>  <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/raster/man/as.matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">vegan</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html">vegdist</a></span><span class="op">(</span><span class="va">x</span>, binary<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="va">fun_BAT</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/raster/man/as.matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">BAT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BAT/man/beta.html">beta</a></span><span class="op">(</span><span class="va">x</span>, func <span class="op">=</span> <span class="st">"Soerensen"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="va">res3</span> <span class="op">&lt;-</span> <span class="fu">bench</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bench/man/mark.html">mark</a></span><span class="op">(</span>phyloregion<span class="op">=</span><span class="fu">fun_phyloregion</span><span class="op">(</span><span class="va">X_sparse</span><span class="op">)</span>,
                    betapart<span class="op">=</span><span class="fu">fun_betapart</span><span class="op">(</span><span class="va">X_dense</span><span class="op">)</span>,
                    vegan<span class="op">=</span><span class="fu">fun_vegan</span><span class="op">(</span><span class="va">X_dense</span><span class="op">)</span>,
                    BAT<span class="op">=</span><span class="fu">fun_BAT</span><span class="op">(</span><span class="va">X_dense</span><span class="op">)</span>, check<span class="op">=</span><span class="va">chk_fun</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">res3</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 4 x 6
##   expression       min   median `itr/sec` mem_alloc `gc/sec`
##   &lt;bch:expr&gt;  &lt;bch:tm&gt; &lt;bch:tm&gt;     &lt;dbl&gt; &lt;bch:byt&gt;    &lt;dbl&gt;
## 1 phyloregion  753.1µs  929.9µs     903.    404.1KB     2.10
## 2 betapart     863.1µs    924µs     998.    594.1KB     4.38
## 3 vegan        945.8µs      1ms     946.    993.4KB     7.05
## 4 BAT           38.3ms   44.3ms      23.6    31.8MB    15.7</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html">autoplot</a></span><span class="op">(</span><span class="va">res3</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="benchmark-beta_diversity-1.png" alt=""><p class="caption">plot of chunk beta_diversity</p>
</div>
</div>
<div id="benchmarking-phyloregion-for-analysis-of-phylogenetic-beta-diversity" class="section level4">
<h4 class="hasAnchor">
<a href="#benchmarking-phyloregion-for-analysis-of-phylogenetic-beta-diversity" class="anchor"></a>2.2. Benchmarking <code>phyloregion</code> for analysis of phylogenetic beta diversity</h4>
<p>For analysis of phylogenetic turnover (beta-diversity) among communities - the proportion of shared phylogenetic branch lengths between communities<span class="citation"><sup>8</sup></span> - <code>phyloregion</code> is 300-400 times faster and allocates 100-600 times less memory!</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fun_phyloregion</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">phyloregion</span><span class="fu">::</span><span class="fu"><a href="../reference/phylobeta_core.html">phylobeta</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span>
<span class="va">fun_betapart</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">betapart</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/betapart/man/phylo.beta.pair.html">phylo.beta.pair</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span>
<span class="va">fun_picante</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fl">1</span> <span class="op">-</span> <span class="fu">picante</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/picante/man/phylosor.html">phylosor</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span>
<span class="va">fun_BAT</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span><span class="op">)</span> <span class="fu">BAT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BAT/man/beta.html">beta</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">tree</span>, func <span class="op">=</span> <span class="st">"Soerensen"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>

<span class="va">chk_fun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">target</span>, <span class="va">current</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/all.equal.html">all.equal</a></span><span class="op">(</span><span class="va">target</span>, <span class="va">current</span>, check.attributes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="va">res4</span> <span class="op">&lt;-</span> <span class="fu">bench</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bench/man/mark.html">mark</a></span><span class="op">(</span>picante<span class="op">=</span><span class="fu">fun_picante</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,
                       betapart<span class="op">=</span><span class="fu">fun_betapart</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,
                       BAT<span class="op">=</span><span class="fu">fun_BAT</span><span class="op">(</span><span class="va">X_dense</span>, <span class="va">tree</span><span class="op">)</span>,
                       phyloregion<span class="op">=</span><span class="fu">fun_phyloregion</span><span class="op">(</span><span class="va">X_sparse</span>, <span class="va">tree</span><span class="op">)</span>, check<span class="op">=</span><span class="va">chk_fun</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">res4</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 4 x 6
##   expression       min   median `itr/sec` mem_alloc `gc/sec`
##   &lt;bch:expr&gt;  &lt;bch:tm&gt; &lt;bch:tm&gt;     &lt;dbl&gt; &lt;bch:byt&gt;    &lt;dbl&gt;
## 1 picante        1.86s    1.86s     0.539    1.24GB    1.62 
## 2 betapart       1.83s    1.83s     0.545    1.24GB    2.18 
## 3 BAT            1.42s    1.42s     0.703  208.99MB    0.703
## 4 phyloregion   4.34ms   4.52ms   215.       1.15MB    0</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html">autoplot</a></span><span class="op">(</span><span class="va">res4</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="benchmark-phylobeta-1.png" alt=""><p class="caption">plot of chunk phylobeta</p>
</div>
<p>Note that for this test, <code>picante</code> returns a similarity matrix while <code>betapart</code>, and <code>phyloregion</code> return a dissimilarity matrix.</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.6
## 
## Matrix products: default
## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] pez_1.2-0         BAT_2.0.1         hilldiv_1.5.1     picante_1.8.2    
##  [5] nlme_3.1-148      vegan_2.5-6       lattice_0.20-41   permute_0.9-5    
##  [9] betapart_1.5.1    phyloregion_1.0.4 ggplot2_3.3.2     bench_1.1.1      
## [13] Matrix_1.2-18     ape_5.4           knitr_1.29       
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.1.4              ks_1.11.7               tidyselect_1.1.0       
##   [4] htmlwidgets_1.5.1       grid_4.0.2              combinat_0.0-8         
##   [7] munsell_0.5.0           animation_2.6           codetools_0.2-16       
##  [10] miniUI_0.1.1.1          withr_2.2.0             profmem_0.5.0          
##  [13] colorspace_1.4-1        highr_0.8               rstudioapi_0.11        
##  [16] geometry_0.4.5          stats4_4.0.2            tensor_1.5             
##  [19] ggsignif_0.6.0          huge_1.3.4.1            FD_1.0-12              
##  [22] nls2_0.2                polyclip_1.10-0         mnormt_2.0.1           
##  [25] farver_2.0.3            rprojroot_1.3-2         coda_0.19-3            
##  [28] vctrs_0.3.1             generics_0.0.2          clusterGeneration_1.3.4
##  [31] xfun_0.15               fastcluster_1.1.25      R6_2.4.1               
##  [34] ggbeeswarm_0.6.0        pdist_1.2               manipulateWidget_0.10.1
##  [37] spatstat.utils_1.17-0   assertthat_0.2.1        promises_1.1.1         
##  [40] scales_1.1.1            nnet_7.3-14             beeswarm_0.2.3         
##  [43] rgeos_0.5-3             gtable_0.3.0            caper_1.0.1            
##  [46] goftest_1.2-2           processx_3.4.3          phangorn_2.5.5         
##  [49] MatrixModels_0.4-1      rlang_0.4.6             FSA_0.8.30             
##  [52] scatterplot3d_0.3-41    splines_4.0.2           rstatix_0.6.0          
##  [55] acepack_1.4.1           broom_0.7.0             checkmate_2.0.0        
##  [58] rgl_0.100.54            yaml_2.2.1              reshape2_1.4.4         
##  [61] abind_1.4-5             d3Network_0.5.2.1       crosstalk_1.1.0.1      
##  [64] backports_1.1.8         httpuv_1.5.4            Hmisc_4.4-0            
##  [67] tools_4.0.2             psych_1.9.12.31         lavaan_0.6-6           
##  [70] cubature_2.0.4.1        ellipsis_0.3.1          raster_3.3-7           
##  [73] RColorBrewer_1.1-2      Rcpp_1.0.5              plyr_1.8.6             
##  [76] base64enc_0.1-3         progress_1.2.2          purrr_0.3.4            
##  [79] ps_1.3.3                prettyunits_1.1.1       deldir_0.1-25          
##  [82] ggpubr_0.4.0            rpart_4.1-15            pbapply_1.4-2          
##  [85] deSolve_1.28            qgraph_1.6.5            haven_2.3.1            
##  [88] cluster_2.1.0           fs_1.4.2                magrittr_1.5           
##  [91] data.table_1.12.8       openxlsx_4.1.5          SparseM_1.78           
##  [94] tmvnsim_1.0-2           mvtnorm_1.1-1           whisker_0.4            
##  [97] hms_0.5.3               mime_0.9                evaluate_0.14          
## [100] xtable_1.8-4            rio_0.5.16              jpeg_0.1-8.1           
## [103] mclust_5.4.6            readxl_1.3.1            gridExtra_2.3          
## [106] compiler_4.0.2          tibble_3.0.2            maps_3.3.0             
## [109] KernSmooth_2.23-17      crayon_1.3.4            hypervolume_2.0.12     
## [112] htmltools_0.5.0         mgcv_1.8-31             corpcor_1.6.9          
## [115] later_1.1.0.1           Formula_1.2-3           tidyr_1.1.0            
## [118] expm_0.999-4            apTreeshape_1.5-0       magic_1.5-9            
## [121] subplex_1.6             MASS_7.3-51.6           ade4_1.7-15            
## [124] car_3.0-8               cli_2.0.2               quadprog_1.5-8         
## [127] parallel_4.0.2          igraph_1.2.5            BDgraph_2.62           
## [130] forcats_0.5.0           pkgconfig_2.0.3         pkgdown_1.5.1          
## [133] numDeriv_2016.8-1.1     foreign_0.8-80          sp_1.4-2               
## [136] pbivnorm_0.6.0          vipor_0.4.5             webshot_0.5.2          
## [139] stringr_1.4.0           callr_3.4.3             digest_0.6.25          
## [142] phytools_0.7-47         rcdd_1.2-2              spatstat.data_1.4-3    
## [145] rmarkdown_2.3           cellranger_1.1.0        fastmatch_1.1-0        
## [148] htmlTable_2.0.1         curl_4.3                quantreg_5.55          
## [151] shiny_1.5.0             gtools_3.8.2            rjson_0.2.20           
## [154] geiger_2.0.7            lifecycle_0.2.0         glasso_1.11            
## [157] jsonlite_1.7.0          carData_3.0-4           desc_1.2.0             
## [160] fansi_0.4.1             pillar_1.4.6            fastmap_1.0.1          
## [163] httr_1.4.1              plotrix_3.7-8           survival_3.2-3         
## [166] glue_1.4.1              spatstat_1.64-1         zip_2.0.4              
## [169] fdrtool_1.2.15          png_0.1-7               class_7.3-17           
## [172] stringi_1.4.6           rematch2_2.1.2          latticeExtra_0.6-29    
## [175] memoise_1.1.0           dplyr_1.0.0             e1071_1.7-3</code></pre>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>REFERENCES</h2>
<div id="refs" class="references">
<div id="ref-Daru2015ddi">
<p>1. Daru, B. H., Bank, M. van der &amp; Davies, T. J. Spatial incongruence among hotspots and complementary areas of tree diversity in southern africa. <em>Diversity and Distributions</em> <strong>21</strong>, 769–780 (2015).</p>
</div>
<div id="ref-Kembel2010">
<p>2. Kembel, S. W. <em>et al.</em> Picante: R tools for integrating phylogenies and ecology. <em>Bioinformatics</em> <strong>26</strong>, 1463–1464 (2010).</p>
</div>
<div id="ref-hilldiv">
<p>3. Alberdi, A. Hilldiv: An r package for the integral analysis of diversity based on hill numbers. <em>bioRxiv</em> (2019).</p>
</div>
<div id="ref-bench2020">
<p>4. Hester, J. <em>Bench: High precision timing of r expressions</em>. (2020).</p>
</div>
<div id="ref-Faith1992">
<p>5. Faith, D. P. Conservation evaluation and phylogenetic diversity. <em>Biological Conservation</em> <strong>61</strong>, 1–10 (1992).</p>
</div>
<div id="ref-Rosauer2009">
<p>6. Rosauer, D., Laffan, S. W., Crisp, M. D., Donnellan, S. C. &amp; Cook, L. G. Phylogenetic endemism: A new approach for identifying geographical concentrations of evolutionary history. <em>Molecular Ecology</em> <strong>18</strong>, 4061–4072 (2009).</p>
</div>
<div id="ref-Koleff2003">
<p>7. Koleff, P., Gaston, K. J. &amp; Lennon, J. J. Measuring beta diversity for presence–absence data. <em>Journal of Animal Ecology</em> <strong>72</strong>, 367–382 (2003).</p>
</div>
<div id="ref-Graham2008">
<p>8. Graham, C. H. &amp; Fine, P. V. A. Phylogenetic beta diversity: Linking ecological and evolutionary processes across space in time. <em>Ecology Letters</em> <strong>11</strong>, 1265–1277 (2008).</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Barnabas H. Daru, Piyal Karunarathne, Klaus Schliep.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
