<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction to the phyloregion package • phyloregion</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to the phyloregion package">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phyloregion</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Benchmark.html">Benchmarking phyloregion</a>
    </li>
    <li>
      <a href="../articles/phyloregion-intro.html">An introduction to the phyloregion package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/darunabas/phyloregion">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>An introduction to the phyloregion package</h1>
                        <h4 class="author">Barnabas H. Daru, Piyal Karunarathne &amp; Klaus Schliep</h4>
            
            <h4 class="date">März 09, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/darunabas/phyloregion/blob/master/vignettes/phyloregion-intro.Rmd"><code>vignettes/phyloregion-intro.Rmd</code></a></small>
      <div class="hidden name"><code>phyloregion-intro.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>1. Installation</h2>
<p><code>phyloregion</code> is a free R package hosted on <a href="https://github.com/darunabas/phyloregion">GitHub</a>. To install <code>phyloregion</code>, with the help from the <code>remotes</code> package, type the following commands in R:</p>
<pre><code>if (!requireNamespace("remotes", quietly = TRUE)) 
    install.packages("remotes") 
devtools::install_github("darunabas/phyloregion")</code></pre>
<p>When installed, load the package in R:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(phyloregion)</a></code></pre></div>
<pre><code>## 
## Attache Paket: 'phyloregion'</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     colSums, rowSums</code></pre>
</div>
<div id="input-data" class="section level2">
<h2 class="hasAnchor">
<a href="#input-data" class="anchor"></a>2. Input data</h2>
<div id="phylogenies" class="section level3">
<h3 class="hasAnchor">
<a href="#phylogenies" class="anchor"></a>Phylogenies</h3>
<p>In R, phylogenetic relationships among species / taxa are often represented as a phylo object implemented in the <code>ape</code> package <span class="citation">(Paradis and Schliep 2018)</span>. Phylogenies (often in the Newick or Nexus formats) can be imported into R with the <code>read.tree</code> or <code>read.nexus</code> functions of the <code>ape</code> package.</p>
</div>
<div id="community-data" class="section level3">
<h3 class="hasAnchor">
<a href="#community-data" class="anchor"></a>Community data</h3>
<p>Community data are commonly stored in a matrix with the sites as rows and species / operational taxonomic units (OTUs) as columns. The elements of the matrix are numeric values indicating the abundance/observations or presence/absence (0/1) of OTUs in different sites. In practice, such a matrix can contain many zero values because species are known to generally have unimodal distributions along environmental gradients <span class="citation">(Ter Braak and Prentice 2004)</span>, and storing and analyzing every single element of that matrix can be computationally challenging and expensive.</p>
<p><code>phyloregion</code> differs from other R packages (e.g. vegan <span class="citation">(Oksanen et al. 2019)</span>, picante <span class="citation">(Kembel et al. 2010)</span> or betapart <span class="citation">(Baselga and Orme 2012)</span>) in that the data are not stored in a (dense) <code>matrix</code> or <code>data.frame</code> but as a sparse matrix making use of the infrastructure provided by the Matrix package <span class="citation">(Bates and Maechler 2019)</span>. A sparse matrix is a matrix with a high proportion of zero entries <span class="citation">(Duff 1977)</span>, of which only the non-zero entries are stored and utlized for downstream analysis.</p>
<p>A sparse matrix representation has two advantages. First the community matrix can be stored in a much memory efficient manner, allowing analysis of larger datasets. Second, for very large datasets spanning thousands of taxa and spatial scales, computations with a sparse matrix are often much faster.<br>
The <code>phyloregion</code> package contains functions to conveniently change between data formats.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(Matrix) </a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(africa)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">sparse_comm &lt;-<span class="st"> </span>africa<span class="op">$</span>comm</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">dense_comm &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(sparse_comm) </a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/utils/object.size.html">object.size</a></span>(dense_comm)</a></code></pre></div>
<pre><code>## 4216952 bytes</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/object.size.html">object.size</a></span>(sparse_comm)</a></code></pre></div>
<pre><code>## 885952 bytes</code></pre>
<p>Here, the data set in the dense matrix representation consumes roughly five times more memory than the sparse representation.</p>
</div>
</div>
<div id="analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis" class="anchor"></a>3. Analysis</h2>
<div id="phylogenetic-beta-diversity" class="section level3">
<h3 class="hasAnchor">
<a href="#phylogenetic-beta-diversity" class="anchor"></a>Phylogenetic beta diversity</h3>
<p><code>phyloregion</code> offers a fast means of computing phylogenetic beta diversity, the turnover of branch lengths among sites, making use of and improving on the infrastructure provided by the <code>betapart</code> package <span class="citation">(Baselga and Orme 2012)</span>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">tree &lt;-<span class="st"> </span>africa<span class="op">$</span>phylo</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">tree &lt;-<span class="st"> </span>ape<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/ape/man/drop.tip.html">keep.tip</a></span>(tree, <span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(sparse_comm))</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">pb &lt;-<span class="st"> </span><span class="kw"><a href="../reference/phylobeta.html">phylobeta</a></span>(sparse_comm, tree)</a></code></pre></div>
</div>
</div>
<div id="session-infomation" class="section level2">
<h2 class="hasAnchor">
<a href="#session-infomation" class="anchor"></a>Session Infomation</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a></code></pre></div>
<pre><code>## R version 3.6.3 (2020-02-29)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 19.10
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.8.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.8.0
## 
## locale:
##  [1] LC_CTYPE=de_AT.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=de_AT.UTF-8        LC_COLLATE=de_AT.UTF-8    
##  [5] LC_MONETARY=de_AT.UTF-8    LC_MESSAGES=de_AT.UTF-8   
##  [7] LC_PAPER=de_AT.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=de_AT.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] Matrix_1.2-18     phyloregion_1.0.0
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.3        compiler_3.6.3    tools_3.6.3       magic_1.5-9      
##  [5] digest_0.6.25     betapart_1.5.1    evaluate_0.14     memoise_1.1.0    
##  [9] nlme_3.1-144      lattice_0.20-40   mgcv_1.8-31       pkgconfig_2.0.3  
## [13] rlang_0.4.5       igraph_1.2.4.2    fastmatch_1.1-0   rstudioapi_0.11  
## [17] yaml_2.2.1        parallel_3.6.3    pkgdown_1.4.1     xfun_0.12        
## [21] raster_3.0-12     stringr_1.4.0     knitr_1.28        cluster_2.1.0    
## [25] rgeos_0.5-2       desc_1.2.0        fs_1.3.1          rcdd_1.2-2       
## [29] rprojroot_1.3-2   grid_3.6.3        data.table_1.12.8 R6_2.4.1         
## [33] phangorn_2.5.5    rmarkdown_2.1     sp_1.4-1          magrittr_1.5     
## [37] codetools_0.2-16  splines_3.6.3     backports_1.1.5   htmltools_0.4.0  
## [41] MASS_7.3-51.5     assertthat_0.2.1  abind_1.4-5       colorspace_1.4-1 
## [45] picante_1.8.1     permute_0.9-5     ape_5.3           quadprog_1.5-8   
## [49] stringi_1.4.6     geometry_0.4.5    crayon_1.3.4      vegan_2.5-6</code></pre>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>REFERENCES</h2>
<div id="refs" class="references">
<div id="ref-Baselga2012">
<p>Baselga, Andrés, and C David L Orme. 2012. “Betapart: An R Package for the Study of Beta Diversity.” <em>Methods in Ecology and Evolution</em> 3 (5). Wiley Online Library: 808–12.</p>
</div>
<div id="ref-Matrix">
<p>Bates, Douglas, and Martin Maechler. 2019. <em>Matrix: Sparse and Dense Matrix Classes and Methods</em>. <a href="https://CRAN.R-project.org/package=Matrix">https://CRAN.R-project.org/package=Matrix</a>.</p>
</div>
<div id="ref-Daru2017">
<p>Daru, Barnabas H, Tammy L Elliott, Daniel S Park, and T Jonathan Davies. 2017. “Understanding the Processes Underpinning Patterns of Phylogenetic Regionalization.” <em>Trends in Ecology &amp; Evolution</em> 32 (11). Elsevier: 845–60.</p>
</div>
<div id="ref-Daru2020">
<p>Daru, Barnabas H., Piyal Karunarathne, and Klaus Schliep. 2020. “Phyloregion: R Package for Biogeographic Regionalization and Spatial Conservation.” <em>bioRxiv</em>. Cold Spring Harbor Laboratory. <a href="https://doi.org/10.1101/2020.02.12.945691">https://doi.org/10.1101/2020.02.12.945691</a>.</p>
</div>
<div id="ref-Duff1977">
<p>Duff, I. S. 1977. “A Survey of Sparse Matrix Research.” <em>Proceedings of the IEEE</em> 65 (4): 500–535. <a href="https://doi.org/10.1109/PROC.1977.10514">https://doi.org/10.1109/PROC.1977.10514</a>.</p>
</div>
<div id="ref-Kembel2010">
<p>Kembel, S.W., P.D. Cowan, M.R. Helmus, W.K. Cornwell, H. Morlon, D.D. Ackerly, S.P. Blomberg, and C.O. Webb. 2010. “Picante: R Tools for Integrating Phylogenies and Ecology.” <em>Bioinformatics</em> 26: 1463–4.</p>
</div>
<div id="ref-vegan">
<p>Oksanen, Jari, F. Guillaume Blanchet, Michael Friendly, Roeland Kindt, Pierre Legendre, Dan McGlinn, Peter R. Minchin, et al. 2019. <em>Vegan: Community Ecology Package</em>. <a href="https://CRAN.R-project.org/package=vegan">https://CRAN.R-project.org/package=vegan</a>.</p>
</div>
<div id="ref-Paradis2018">
<p>Paradis, E., and K. Schliep. 2018. “Ape 5.0: An Environment for Modern Phylogenetics and Evolutionary Analyses in R.” <em>Bioinformatics</em> 35: 526–28.</p>
</div>
<div id="ref-TerBraak2004">
<p>Ter Braak, Cajo J.F., and I.Colin Prentice. 2004. “A Theory of Gradient Analysis.” In <em>Advances in Ecological Research: Classic Papers</em>, 34:235–82. Advances in Ecological Research. Academic Press. <a href="https://doi.org/https://doi.org/10.1016/S0065-2504(03)34003-6">https://doi.org/https://doi.org/10.1016/S0065-2504(03)34003-6</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation">1. Installation</a></li>
      <li><a href="#input-data">2. Input data</a></li>
      <li><a href="#analysis">3. Analysis</a></li>
      <li><a href="#session-infomation">Session Infomation</a></li>
      <li><a href="#references">REFERENCES</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Barnabas H. Daru, Piyal Karunarathne, Klaus Schliep.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
