<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction to the phyloregion package • phyloregion</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to the phyloregion package">
<meta property="og:description" content="phyloregion">
<meta property="og:image" content="https://phyloregion.com/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phyloregion</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/phyloregion-intro.html">Vig 1: An introduction to the phyloregion package</a>
    </li>
    <li>
      <a href="../articles/Benchmark.html">Vig 2: Benchmarking phyloregion</a>
    </li>
    <li>
      <a href="../articles/ed.html">Vig 3: Fast Evolutionary Distinctiveness</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/darunabas/phyloregion/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An introduction to the phyloregion package</h1>
                        <h4 data-toc-skip class="author">Barnabas H.
Daru, Piyal Karunarathne &amp; Klaus Schliep</h4>
            
            <h4 data-toc-skip class="date">January 27, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/darunabas/phyloregion/blob/HEAD/vignettes/phyloregion-intro.Rmd" class="external-link"><code>vignettes/phyloregion-intro.Rmd</code></a></small>
      <div class="hidden name"><code>phyloregion-intro.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="installation">1. Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><code>phyloregion</code> is available from the <a href="https://CRAN.R-project.org/package=phyloregion" class="external-link">Comprehensive R
Archive Network</a>, so you can use the following line of code to
install and run it:</p>
<pre><code><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"phyloregion"</span><span class="op">)</span></span></code></pre>
<p>Alternatively, you can install the development version of
<code>phyloregion</code> hosted on GitHub. To do this, you will need to
install the <code>devtools</code> package. In R, type:</p>
<pre><code><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"remotes"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span> </span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"darunabas/phyloregion"</span><span class="op">)</span></span></code></pre>
<p>When installed, load the package in R:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/darunabas/phyloregion" class="external-link">phyloregion</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="overview-and-general-workflow-of-phyloregion">2. Overview and general workflow of <code>phyloregion</code><a class="anchor" aria-label="anchor" href="#overview-and-general-workflow-of-phyloregion"></a>
</h2>
<p>The workflow of the <code>phyloregion</code> package demonstrates
steps from preparation of different types of data to visualizing the
results of biogeographical regionalization, together with tips on
selecting the optimal method for achieving the best output, depending on
the types of data used and research questions.</p>
<div class="float">
<img src="workflow.png" alt="Figure 1. Simplified workflow for analysis of biogeographical regionalization using phyloregion. Distribution data is converted to a sparse community matrix. When paired with phylogenetic data, phylobuilder creates a subtree with largest overlap from a species list, thereby ensuring complete representation of missing data; phylocommunity matrix to visualization of results."><div class="figcaption">
<strong>Figure 1.</strong> Simplified workflow
for analysis of biogeographical regionalization using phyloregion.
Distribution data is converted to a sparse community matrix. When paired
with phylogenetic data, phylobuilder creates a subtree with largest
overlap from a species list, thereby ensuring complete representation of
missing data; phylocommunity matrix to visualization of results.</div>
</div>
</div>
<div class="section level2">
<h2 id="input-data">3. Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<div class="section level3">
<h3 id="phylogenies">Phylogenies<a class="anchor" aria-label="anchor" href="#phylogenies"></a>
</h3>
<p>In R, phylogenetic relationships among species / taxa are often
represented as a phylo object implemented in the <code>ape</code>
package<span class="citation">(Paradis and Schliep 2018)</span>.
Phylogenies (often in the Newick or Nexus formats) can be imported into
R with the <code>read.tree</code> or <code>read.nexus</code> functions
of the <code>ape</code> package<span class="citation">(Paradis and
Schliep 2018)</span>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/" class="external-link">ape</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://Matrix.R-forge.R-project.org" class="external-link">Matrix</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rspatial.org/" class="external-link">terra</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## terra 1.7.55</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'terra'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:ape':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rotate, trans, zoom</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">africa</span><span class="op">)</span></span>
<span><span class="va">sparse_comm</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">comm</span></span>
<span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">phylo</span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/drop.tip.html" class="external-link">keep.tip</a></span><span class="op">(</span><span class="va">tree</span>, <span class="fu"><a href="https://rdrr.io/pkg/terra/man/intersect.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sparse_comm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tree</span>, show.tip.label<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-2-1.png" alt="__Figure 2.__ Phylogenetic tree of the woody plants of southern Africa inferred from DNA barcodes using a maximum likelihood approach and transforming branch lengths to millions of years ago by enforcing a relaxed molecular clock and multiple calibrations [@Daru2015ddi]." width="700"><p class="caption">
<strong>Figure 2.</strong> Phylogenetic tree of the woody plants of
southern Africa inferred from DNA barcodes using a maximum likelihood
approach and transforming branch lengths to millions of years ago by
enforcing a relaxed molecular clock and multiple calibrations <span class="citation">(Barnabas H. Daru, Bank, and Davies 2015)</span>.
</p>
</div>
</div>
<div class="section level3">
<h3 id="distribution-data-input">Distribution data input<a class="anchor" aria-label="anchor" href="#distribution-data-input"></a>
</h3>
<p>The <code>phyloregion</code> package has functions for manipulating
three kinds of distribution data: point records, vector polygons and
raster layers. An overview can be easily obtained with the functions
<code>points2comm</code>, <code>polys2comm</code> and
<code>rast2comm</code> for point records, polygons, or raster layers,
respectively. Depending on the data source, all three functions
ultimately provide convenient interfaces to convert the distribution
data to a community matrix at varying spatial grains and extents for
downstream analyses.</p>
<p>We will play around with these functions in turn.</p>
</div>
<div class="section level3">
<h3 id="function-points2comm">
<strong><em>Function <code>points2comm</code></em></strong><a class="anchor" aria-label="anchor" href="#function-points2comm"></a>
</h3>
<p>Here, we will generate random points in geographic space, similar to
occurrence data obtained from museum records, GBIF, iDigBio, or CIESIN
which typically have columns of geographic coordinates for each
observation.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/vect.html" class="external-link">vect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/nigeria.json"</span>, package<span class="op">=</span><span class="st">"phyloregion"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/sample.html" class="external-link">spatSample</a></span><span class="op">(</span><span class="va">s</span>, <span class="fl">1000</span>, method <span class="op">=</span> <span class="st">"random"</span><span class="op">)</span>,</span>
<span>                   geom <span class="op">=</span> <span class="st">"XY"</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lon"</span>, <span class="st">"lat"</span><span class="op">)</span></span>
<span><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"sp"</span>, <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">m</span><span class="op">$</span><span class="va">taxon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">species</span>, size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/dimensions.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rast2comm.html">points2comm</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">m</span>, res <span class="op">=</span> <span class="fl">0.5</span>, lon <span class="op">=</span> <span class="st">"lon"</span>, lat <span class="op">=</span> <span class="st">"lat"</span>,</span>
<span>            species <span class="op">=</span> <span class="st">"taxon"</span><span class="op">)</span> <span class="co"># This generates a list of two objects</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/headtail.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pt</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 5 x 5 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##      sp1 sp10 sp100 sp11 sp12</span></span>
<span><span class="co">## v100   .    .     .    .    .</span></span>
<span><span class="co">## v101   .    1     .    .    .</span></span>
<span><span class="co">## v102   .    .     .    .    .</span></span>
<span><span class="co">## v103   .    .     .    .    .</span></span>
<span><span class="co">## v105   .    .     .    .    .</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="function-polys2comm">
<strong><em>Function <code>polys2comm</code></em></strong><a class="anchor" aria-label="anchor" href="#function-polys2comm"></a>
</h3>
<p>This function converts polygons to a community matrix at varying
spatial grains and extents for downstream analyses. Polygons can be
derived from the IUCN Redlist spatial database (https:
//www.iucnredlist.org/resources/spatial-data-download), published
monographs or field guides validated by taxonomic experts. To illustrate
this function, we will use the function <code>random_species</code> to
generate random polygons for five random species over the landscape of
Nigeria as follows:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/vect.html" class="external-link">vect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/nigeria.json"</span>, package<span class="op">=</span><span class="st">"phyloregion"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/random_species.html">random_species</a></span><span class="op">(</span><span class="fl">100</span>, species<span class="op">=</span><span class="fl">5</span>, pol<span class="op">=</span><span class="va">s</span><span class="op">)</span></span>
<span><span class="va">pol</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rast2comm.html">polys2comm</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">sp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/headtail.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pol</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 5 x 5 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##       species1 species2 species3 species4 species5</span></span>
<span><span class="co">## v10          .        .        .        1        .</span></span>
<span><span class="co">## v100         .        .        .        1        1</span></span>
<span><span class="co">## v1000        .        .        .        1        1</span></span>
<span><span class="co">## v1001        .        .        .        1        1</span></span>
<span><span class="co">## v1002        .        .        .        1        1</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="function-rast2comm">
<strong><em>Function <code>rast2comm</code></em></strong><a class="anchor" aria-label="anchor" href="#function-rast2comm"></a>
</h3>
<p>This third function, converts raster layers (often derived from
species distribution modeling, such as aquamaps<span class="citation">(Kaschner et al. 2008)</span>) to a community
matrix.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fdir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"NGAplants"</span>, package<span class="op">=</span><span class="st">"phyloregion"</span><span class="op">)</span></span>
<span><span class="va">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">fdir</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="va">fdir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ras</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rast2comm.html">rast2comm</a></span><span class="op">(</span><span class="va">files</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/headtail.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ras</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 5 x 5 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##      Chytranthus_gilletii Commelina_ramulosa Cymbopogon_caesius</span></span>
<span><span class="co">## v100                    .                  .                  .</span></span>
<span><span class="co">## v101                    .                  .                  .</span></span>
<span><span class="co">## v102                    .                  .                  .</span></span>
<span><span class="co">## v103                    .                  .                  .</span></span>
<span><span class="co">## v104                    .                  .                  .</span></span>
<span><span class="co">##      Dalechampia_ipomoeifolia Grewia_barombiensis</span></span>
<span><span class="co">## v100                        1                   .</span></span>
<span><span class="co">## v101                        1                   .</span></span>
<span><span class="co">## v102                        1                   .</span></span>
<span><span class="co">## v103                        1                   .</span></span>
<span><span class="co">## v104                        1                   .</span></span></code></pre>
<p>The object <code>ras</code> above also returns two objects: a
community data frame and a vector of grid cells with the numbers of
species per cell and can be plotted as a heatmap using <code>plot</code>
function as follows:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/vect.html" class="external-link">vect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/SR_Naija.json"</span>, package<span class="op">=</span><span class="st">"phyloregion"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">s</span>, <span class="st">"SR"</span>, border<span class="op">=</span><span class="cn">NA</span>, type <span class="op">=</span> <span class="st">"continuous"</span>, </span>
<span>     col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">hcl.colors</a></span><span class="op">(</span><span class="fl">20</span>, palette <span class="op">=</span> <span class="st">"Blue-Red 3"</span>, rev<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-6-1.png" alt="__Figure 3.__ Species richness of plants in Nigeria across equal area grid cells. This is to demonstrate how the function `plot` works." width="50%"><p class="caption">
<strong>Figure 3.</strong> Species richness of plants in Nigeria across
equal area grid cells. This is to demonstrate how the function
<code>plot</code> works.
</p>
</div>
</div>
<div class="section level3">
<h3 id="community-data">Community data<a class="anchor" aria-label="anchor" href="#community-data"></a>
</h3>
<p>Community data are commonly stored in a matrix with the sites as rows
and species / operational taxonomic units (OTUs) as columns. The
elements of the matrix are numeric values indicating the
abundance/observations or presence/absence (0/1) of OTUs in different
sites. In practice, such a matrix can contain many zero values because
species are known to generally have unimodal distributions along
environmental gradients <span class="citation">(Ter Braak and Prentice
2004)</span>, and storing and analyzing every single element of that
matrix can be computationally challenging and expensive.</p>
<p><code>phyloregion</code> differs from other R packages (e.g. vegan
<span class="citation">(Oksanen et al. 2019)</span>, picante <span class="citation">(Kembel et al. 2010)</span> or betapart<span class="citation">(Baselga and Orme 2012)</span>) in that the data are
not stored in a (dense) <code>matrix</code> or <code>data.frame</code>
but as a sparse matrix making use of the infrastructure provided by the
Matrix package <span class="citation">(Bates and Maechler 2019)</span>.
A sparse matrix is a matrix with a high proportion of zero entries<span class="citation">(Duff 1977)</span>, of which only the non-zero entries
are stored and used for downstream analysis.</p>
<p>A sparse matrix representation has two advantages. First the
community matrix can be stored in a much memory efficient manner,
allowing analysis of larger datasets. Second, for very large datasets
spanning thousands of taxa and spatial scales, computations with a
sparse matrix are often much faster.<br>
The <code>phyloregion</code> package contains functions to conveniently
change between data formats.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://Matrix.R-forge.R-project.org" class="external-link">Matrix</a></span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">africa</span><span class="op">)</span></span>
<span><span class="va">sparse_comm</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">comm</span></span>
<span><span class="va">dense_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/coerce.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">sparse_comm</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/object.size.html" class="external-link">object.size</a></span><span class="op">(</span><span class="va">dense_comm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 4216952 bytes</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/object.size.html" class="external-link">object.size</a></span><span class="op">(</span><span class="va">sparse_comm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 885952 bytes</span></span></code></pre>
<p>Here, the data set in the dense matrix representation consumes
roughly five times more memory than the sparse representation.</p>
</div>
</div>
<div class="section level2">
<h2 id="analysis">4. Analysis<a class="anchor" aria-label="anchor" href="#analysis"></a>
</h2>
<div class="section level3">
<h3 id="alpha-diversity">Alpha diversity<a class="anchor" aria-label="anchor" href="#alpha-diversity"></a>
</h3>
<p>We demonstrate the utility of <code>phyloregion</code> in mapping
standard conservation metrics of species richness, weighted endemism
(<code>weighted_endemism</code>) and threat (<code>map_traits</code>) as
well as fast computations of phylodiversity measures such as
phylogenetic diversity (<code>PD</code>), phylogenetic endemism
(<code>phylo_endemism</code>), and evolutionary distinctiveness and
global endangerment (<code>EDGE</code>). The major advantage of these
functions compared to available tools is the ability to utilize sparse
matrix that speeds up the analyses without exhausting computer memories,
making it ideal for handling any data from small local scales to large
regional and global scales.</p>
<div class="section level4">
<h4 id="function-weighted_endemism">Function <code>weighted_endemism</code><a class="anchor" aria-label="anchor" href="#function-weighted_endemism"></a>
</h4>
<p>Weighted endemism is species richness inversely weighted by species
ranges<span class="citation">(Crisp et al. 2001)</span>,<span class="citation">(Laffan and Crisp 2003)</span>,<span class="citation">(Barnabas H. Daru et al. 2020)</span>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rspatial.org/" class="external-link">terra</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">africa</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/vect.html" class="external-link">vect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/sa.json"</span>, package <span class="op">=</span> <span class="st">"phyloregion"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">Endm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/weighted_endemism.html">weighted_endemism</a></span><span class="op">(</span><span class="va">africa</span><span class="op">$</span><span class="va">comm</span><span class="op">)</span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">p</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>grids<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Endm</span><span class="op">)</span>, WE<span class="op">=</span><span class="va">Endm</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"grids"</span><span class="op">)</span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="va">m</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">m</span><span class="op">$</span><span class="va">WE</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">m</span>, <span class="st">"WE"</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">hcl.colors</a></span><span class="op">(</span><span class="fl">20</span>, <span class="st">"Blue-Red 3"</span><span class="op">)</span>, </span>
<span>     type<span class="op">=</span><span class="st">"continuous"</span>, border <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-8-1.png" alt="__Figure 4.__ Geographic distributions of weighted endemism for woody plants of southern Africa." width="50%"><p class="caption">
<strong>Figure 4.</strong> Geographic distributions of weighted endemism
for woody plants of southern Africa.
</p>
</div>
</div>
<div class="section level4">
<h4 id="function-pd-phylogenetic-diversity">Function <code>PD</code> – phylogenetic diversity<a class="anchor" aria-label="anchor" href="#function-pd-phylogenetic-diversity"></a>
</h4>
<p>Phylogenetic diversity (<code>PD</code>) represents the length of
evolutionary pathways that connects a given set of taxa on a rooted
phylogenetic tree <span class="citation">(Faith 1992)</span>. This
metric is often characterised in units of time (millions of years, for
dated phylogenies). We will map PD for plants of southern Africa.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">africa</span><span class="op">)</span></span>
<span><span class="va">comm</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">comm</span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">phylo</span></span>
<span><span class="va">poly</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/vect.html" class="external-link">vect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/sa.json"</span>, package <span class="op">=</span> <span class="st">"phyloregion"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mypd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PD.html">PD</a></span><span class="op">(</span><span class="va">comm</span>, <span class="va">tree</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/headtail.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mypd</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    v3635    v3636    v3637    v3638    v3639    v3640 </span></span>
<span><span class="co">## 4226.216 5372.009 4377.735 3783.992 3260.111 1032.685</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">M</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">poly</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>grids<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">mypd</span><span class="op">)</span>, pd<span class="op">=</span><span class="va">mypd</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"grids"</span><span class="op">)</span></span>
<span><span class="va">M</span> <span class="op">&lt;-</span> <span class="va">M</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">M</span><span class="op">$</span><span class="va">pd</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/headtail.html" class="external-link">head</a></span><span class="op">(</span><span class="va">M</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   grids       pd</span></span>
<span><span class="co">## 1 v3635 4226.216</span></span>
<span><span class="co">## 2 v3636 5372.009</span></span>
<span><span class="co">## 3 v3637 4377.735</span></span>
<span><span class="co">## 4 v3638 3783.992</span></span>
<span><span class="co">## 5 v3639 3260.111</span></span>
<span><span class="co">## 6 v3640 1032.685</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">M</span>, <span class="st">"pd"</span>, border<span class="op">=</span><span class="cn">NA</span>, type<span class="op">=</span><span class="st">"continuous"</span>,</span>
<span>            col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">hcl.colors</a></span><span class="op">(</span><span class="fl">20</span>, <span class="st">"Blue-Red 3"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-9-1.png" alt="__Figure 5.__ Geographic distributions of phylogenetic diversity for woody plants of southern Africa." width="50%"><p class="caption">
<strong>Figure 5.</strong> Geographic distributions of phylogenetic
diversity for woody plants of southern Africa.
</p>
</div>
</div>
<div class="section level4">
<h4 id="function-phylo_endemism-phylogenetic-endemism">Function <code>phylo_endemism</code> – phylogenetic endemism<a class="anchor" aria-label="anchor" href="#function-phylo_endemism-phylogenetic-endemism"></a>
</h4>
<p>Phylogenetic endemism is not influenced by variations in taxonomic
opinion because it measures endemism based on the relatedness of species
before weighting it by their range sizes<span class="citation">(Rosauer
et al. 2009)</span>,<span class="citation">(Barnabas H. Daru et al.
2020)</span>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rspatial.org/" class="external-link">terra</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">africa</span><span class="op">)</span></span>
<span><span class="va">comm</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">comm</span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">phylo</span></span>
<span><span class="va">poly</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/vect.html" class="external-link">vect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/sa.json"</span>, package <span class="op">=</span> <span class="st">"phyloregion"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/phylo_endemism.html">phylo_endemism</a></span><span class="op">(</span><span class="va">comm</span>, <span class="va">tree</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/headtail.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pe</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     v3635     v3636     v3637     v3638     v3639     v3640 </span></span>
<span><span class="co">## 32.536530 45.262625 35.004944 27.603721 23.183947  6.439589</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">poly</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>grids<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pe</span><span class="op">)</span>, pe<span class="op">=</span><span class="va">pe</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"grids"</span><span class="op">)</span></span>
<span><span class="va">mx</span> <span class="op">&lt;-</span> <span class="va">mx</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">mx</span><span class="op">$</span><span class="va">pe</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/headtail.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mx</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   grids        pe</span></span>
<span><span class="co">## 1 v3635 32.536530</span></span>
<span><span class="co">## 2 v3636 45.262625</span></span>
<span><span class="co">## 3 v3637 35.004944</span></span>
<span><span class="co">## 4 v3638 27.603721</span></span>
<span><span class="co">## 5 v3639 23.183947</span></span>
<span><span class="co">## 6 v3640  6.439589</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mx</span>, <span class="st">"pe"</span>, border<span class="op">=</span><span class="cn">NA</span>, type<span class="op">=</span><span class="st">"continuous"</span>,</span>
<span>            col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">hcl.colors</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">20</span>, palette <span class="op">=</span> <span class="st">"Blue-Red 3"</span>, rev<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-10-1.png" alt="__Figure 6.__ Geographic distributions of phylogenetic endemism for woody plants of southern Africa." width="50%"><p class="caption">
<strong>Figure 6.</strong> Geographic distributions of phylogenetic
endemism for woody plants of southern Africa.
</p>
</div>
</div>
<div class="section level4">
<h4 id="function-edge-evolutionary-distinctiveness-and-global-endangerment">Function <code>EDGE</code> – Evolutionary Distinctiveness and Global
Endangerment<a class="anchor" aria-label="anchor" href="#function-edge-evolutionary-distinctiveness-and-global-endangerment"></a>
</h4>
<p>This function calculates EDGE by combining evolutionary
distinctiveness (ED; i.e., phylogenetic isolation of a species) with
global endangerment (GE) status as defined by the International Union
for Conservation of Nature (IUCN).</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">africa</span><span class="op">)</span></span>
<span><span class="va">comm</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">comm</span></span>
<span><span class="va">threat</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">IUCN</span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">phylo</span></span>
<span><span class="va">poly</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/vect.html" class="external-link">vect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/sa.json"</span>, package <span class="op">=</span> <span class="st">"phyloregion"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/EDGE.html">EDGE</a></span><span class="op">(</span><span class="va">threat</span>, <span class="va">tree</span>, Redlist <span class="op">=</span> <span class="st">"IUCN"</span>, species<span class="op">=</span><span class="st">"Species"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/headtail.html" class="external-link">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        Abutilon_angulatum_OM1934    Abutilon_sonneratianum_LTM034 </span></span>
<span><span class="co">##                         2.903551                         2.903551 </span></span>
<span><span class="co">## Acalypha_glabrata_glabrata_OM441  Acalypha_glabrata_pilosa_OM1979 </span></span>
<span><span class="co">##                         2.480505                         2.480505 </span></span>
<span><span class="co">##       Acalypha_sonderiana_OM2163  Acokanthera_oblongifolia_OM2240 </span></span>
<span><span class="co">##                         2.914481                         2.211561</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/map_trait.html">map_trait</a></span><span class="op">(</span><span class="va">comm</span>, <span class="va">x</span>, FUN <span class="op">=</span> <span class="va">sd</span>, pol<span class="op">=</span><span class="va">poly</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">y</span>, <span class="st">"traits"</span>, border<span class="op">=</span><span class="cn">NA</span>, type<span class="op">=</span><span class="st">"continuous"</span>,</span>
<span>            col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">hcl.colors</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">20</span>, palette <span class="op">=</span> <span class="st">"Blue-Red 3"</span>, rev<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-11-1.png" alt="__Figure 7.__ Geographic distributions of evolutionary distinctiveness and global endangerment for woody plants of southern Africa." width="50%"><p class="caption">
<strong>Figure 7.</strong> Geographic distributions of evolutionary
distinctiveness and global endangerment for woody plants of southern
Africa.
</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="analysis-of-beta-diversity-phylogenetic-and-non-phylogenetic">Analysis of beta diversity (phylogenetic and non-phylogenetic)<a class="anchor" aria-label="anchor" href="#analysis-of-beta-diversity-phylogenetic-and-non-phylogenetic"></a>
</h3>
<p>The three commonly used methods for quantifying -diversity, the
variation in species composition among sites, – Simpson, Sorenson and
Jaccard<span class="citation">(Laffan et al. 2016)</span>. The
<code>phyloregion</code>’s functions <code>beta_diss</code> and
<code>phylobeta</code> compute efficiently pairwise dissimilarities
matrices for large sparse community matrices and phylogenetic trees for
taxonomic and phylogenetic turnover, respectively. The results are
stored as distance objects for subsequent analyses.</p>
<hr>
</div>
<div class="section level3">
<h3 id="phylogenetic-beta-diversity">Phylogenetic beta diversity<a class="anchor" aria-label="anchor" href="#phylogenetic-beta-diversity"></a>
</h3>
<p><code>phyloregion</code> offers a fast means of computing
phylogenetic beta diversity, the turnover of branch lengths among sites,
making use of and improving on the infrastructure provided by the
<code>betapart</code> package<span class="citation">(Baselga and Orme
2012)</span> allowing a sparse community matrix as input.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">africa</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/vect.html" class="external-link">vect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"ex/sa.json"</span>, package <span class="op">=</span> <span class="st">"phyloregion"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sparse_comm</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">comm</span></span>
<span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="va">africa</span><span class="op">$</span><span class="va">phylo</span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/drop.tip.html" class="external-link">keep.tip</a></span><span class="op">(</span><span class="va">tree</span>, <span class="fu"><a href="https://rdrr.io/pkg/terra/man/intersect.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sparse_comm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/phylobeta.html">phylobeta</a></span><span class="op">(</span><span class="va">sparse_comm</span>, <span class="va">tree</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/phyloregion.html">phyloregion</a></span><span class="op">(</span><span class="va">pb</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, pol<span class="op">=</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot.phyloregion.html">plot_NMDS</a></span><span class="op">(</span><span class="va">y</span>, cex<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in mixcolor(seq(0, 199)/199, polarLUV(70, 50, 30), polarLUV(70, :</span></span>
<span><span class="co">## convex combination of colors in polar coordinates (polarLUV) may not be</span></span>
<span><span class="co">## appropriate</span></span></code></pre>
<pre><code><span><span class="co">## Warning in mixcolor(seq(0, 199)/199, polarLUV(70, 50, 300), polarLUV(70, :</span></span>
<span><span class="co">## convex combination of colors in polar coordinates (polarLUV) may not be</span></span>
<span><span class="co">## appropriate</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot.phyloregion.html">text_NMDS</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span></code></pre></div>
<p><img src="phyloregion-intro_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">y</span>, palette<span class="op">=</span><span class="st">"NMDS"</span><span class="op">)</span></span></code></pre></div>
<p><img src="phyloregion-intro_files/figure-html/unnamed-chunk-14-2.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Sonoma 14.3</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/Los_Angeles</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] terra_1.7-55      Matrix_1.6-1.1    ape_5.7-1         phyloregion_1.0.9</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] fastmatch_1.1-4    xfun_0.41          bslib_0.6.1        clustMixType_0.3-9</span></span>
<span><span class="co">##  [5] lattice_0.21-9     quadprog_1.5-8     vctrs_0.6.5        tools_4.3.2       </span></span>
<span><span class="co">##  [9] doSNOW_1.0.20      generics_0.1.3     parallel_4.3.2     tibble_3.2.1      </span></span>
<span><span class="co">## [13] fansi_1.0.6        cluster_2.1.4      highr_0.10         pkgconfig_2.0.3   </span></span>
<span><span class="co">## [17] RColorBrewer_1.1-3 desc_1.4.2         lifecycle_1.0.4    compiler_4.3.2    </span></span>
<span><span class="co">## [21] stringr_1.5.1      predicts_0.1-11    textshaping_0.3.7  minpack.lm_1.2-4  </span></span>
<span><span class="co">## [25] smoothr_1.0.1      codetools_0.2-19   permute_0.9-7      htmltools_0.5.7   </span></span>
<span><span class="co">## [29] snow_0.4-4         sass_0.4.8         yaml_2.3.7         pillar_1.9.0      </span></span>
<span><span class="co">## [33] pkgdown_2.0.7      jquerylib_0.1.4    MASS_7.3-60        cachem_1.0.8      </span></span>
<span><span class="co">## [37] vegan_2.6-4        iterators_1.0.14   abind_1.4-5        foreach_1.5.2     </span></span>
<span><span class="co">## [41] maptpx_1.9-7       nlme_3.1-163       picante_1.8.2      phangorn_2.11.1   </span></span>
<span><span class="co">## [45] digest_0.6.33      stringi_1.8.2      slam_0.1-50        purrr_1.0.2       </span></span>
<span><span class="co">## [49] splines_4.3.2      magic_1.6-1        rprojroot_2.0.4    fastmap_1.1.1     </span></span>
<span><span class="co">## [53] grid_4.3.2         colorspace_2.1-0   cli_3.6.1          magrittr_2.0.3    </span></span>
<span><span class="co">## [57] utf8_1.2.4         rmarkdown_2.25     igraph_1.5.1       ragg_1.2.6        </span></span>
<span><span class="co">## [61] memoise_2.0.1      evaluate_0.23      knitr_1.45         rcdd_1.5-2        </span></span>
<span><span class="co">## [65] geometry_0.4.7     mgcv_1.9-0         itertools_0.1-3    rlang_1.1.2       </span></span>
<span><span class="co">## [69] Rcpp_1.0.11        glue_1.6.2         betapart_1.6       rstudioapi_0.15.0 </span></span>
<span><span class="co">## [73] jsonlite_1.8.8     R6_2.5.1           systemfonts_1.0.5  fs_1.6.3</span></span></code></pre>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">REFERENCES<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Baselga2012" class="csl-entry">
Baselga, Andrés, and C David L Orme. 2012. <span>“Betapart: An r Package
for the Study of Beta Diversity.”</span> <em>Methods in Ecology and
Evolution</em> 3 (5): 808–12.
</div>
<div id="ref-Matrix" class="csl-entry">
Bates, Douglas, and Martin Maechler. 2019. <em>Matrix: Sparse and Dense
Matrix Classes and Methods</em>. <a href="https://CRAN.R-project.org/package=Matrix" class="external-link">https://CRAN.R-project.org/package=Matrix</a>.
</div>
<div id="ref-crisp2001endemism" class="csl-entry">
Crisp, MICHAEL D, Shawn Laffan, H Peter Linder, and ANNA Monro. 2001.
<span>“Endemism in the Australian Flora.”</span> <em>Journal of
Biogeography</em> 28 (2): 183–98.
</div>
<div id="ref-Daru2015ddi" class="csl-entry">
Daru, Barnabas H., Michelle van der Bank, and T. Jonathan Davies. 2015.
<span>“Spatial Incongruence Among Hotspots and Complementary Areas of
Tree Diversity in Southern Africa.”</span> <em>Diversity and
Distributions</em> 21 (7): 769–80. <a href="https://doi.org/10.1111/ddi.12290" class="external-link">https://doi.org/10.1111/ddi.12290</a>.
</div>
<div id="ref-Daru2017" class="csl-entry">
Daru, Barnabas H, Tammy L Elliott, Daniel S Park, and T Jonathan Davies.
2017. <span>“Understanding the Processes Underpinning Patterns of
Phylogenetic Regionalization.”</span> <em>Trends in Ecology &amp;
Evolution</em> 32 (11): 845–60.
</div>
<div id="ref-daru2020endemism" class="csl-entry">
Daru, Barnabas H., Harith Farooq, Alexandre Antonelli, and Søren Faurby.
2020. <span>“Endemism Patterns Are Scale Dependent.”</span> <em>Nature
Communications</em> 11 (1): 2115. <a href="https://doi.org/10.1038/s41467-020-15921-6" class="external-link">https://doi.org/10.1038/s41467-020-15921-6</a>.
</div>
<div id="ref-Daru2020" class="csl-entry">
Daru, Barnabas H., Piyal Karunarathne, and Klaus Schliep. 2020.
<span>“Phyloregion: R Package for Biogeographic Regionalization and
Macroecology.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/2020.02.12.945691" class="external-link">https://doi.org/10.1101/2020.02.12.945691</a>.
</div>
<div id="ref-Duff1977" class="csl-entry">
Duff, I. S. 1977. <span>“A Survey of Sparse Matrix Research.”</span>
<em>Proceedings of the IEEE</em> 65 (4): 500–535. <a href="https://doi.org/10.1109/PROC.1977.10514" class="external-link">https://doi.org/10.1109/PROC.1977.10514</a>.
</div>
<div id="ref-Faith1992" class="csl-entry">
Faith, Daniel P. 1992. <span>“Conservation Evaluation and Phylogenetic
Diversity.”</span> <em>Biological Conservation</em> 61 (1): 1–10. <a href="https://doi.org/10.1016/0006-3207(92)91201-3" class="external-link">https://doi.org/10.1016/0006-3207(92)91201-3</a>.
</div>
<div id="ref-kaschner2008aquamaps" class="csl-entry">
Kaschner, K, JS Ready, E Agbayani, J Rius, K Kesner-Reyes, PD Eastwood,
AB South, et al. 2008. <span>“AquaMaps: Predicted Range Maps for Aquatic
Species.”</span> <em>World Wide Web Electronic Publication,
Wwwaquamapsorg, Version</em> 10: 2008.
</div>
<div id="ref-Kembel2010" class="csl-entry">
Kembel, S. W., P. D. Cowan, M. R. Helmus, W. K. Cornwell, H. Morlon, D.
D. Ackerly, S. P. Blomberg, and C. O. Webb. 2010. <span>“Picante:
<span>R</span> Tools for Integrating Phylogenies and Ecology.”</span>
<em>Bioinformatics</em> 26: 1463–64.
</div>
<div id="ref-laffan2003assessing" class="csl-entry">
Laffan, Shawn W, and Michael D Crisp. 2003. <span>“Assessing Endemism at
Multiple Spatial Scales, with an Example from the Australian Vascular
Flora.”</span> <em>Journal of Biogeography</em> 30 (4): 511–20.
</div>
<div id="ref-laffan2016range" class="csl-entry">
Laffan, Shawn W, Dan F Rosauer, Giovanni Di Virgilio, Joseph T Miller,
Carlos E González-Orozco, Nunzio Knerr, Andrew H Thornhill, and Brent D
Mishler. 2016. <span>“Range-Weighted Metrics of Species and Phylogenetic
Turnover Can Better Resolve Biogeographic Transition Zones.”</span>
<em>Methods in Ecology and Evolution</em> 7 (5): 580–88.
</div>
<div id="ref-vegan" class="csl-entry">
Oksanen, Jari, F. Guillaume Blanchet, Michael Friendly, Roeland Kindt,
Pierre Legendre, Dan McGlinn, Peter R. Minchin, et al. 2019. <em>Vegan:
Community Ecology Package</em>. <a href="https://CRAN.R-project.org/package=vegan" class="external-link">https://CRAN.R-project.org/package=vegan</a>.
</div>
<div id="ref-Paradis2018" class="csl-entry">
Paradis, E., and K. Schliep. 2018. <span>“Ape 5.0: An Environment for
Modern Phylogenetics and Evolutionary Analyses in
<span>R</span>.”</span> <em>Bioinformatics</em> 35: 526–28.
</div>
<div id="ref-Rosauer2009" class="csl-entry">
Rosauer, Dan, Shawn W. Laffan, Michael D. Crisp, Stephen C. Donnellan,
and Lyn G. Cook. 2009. <span>“Phylogenetic Endemism: A New Approach for
Identifying Geographical Concentrations of Evolutionary History.”</span>
<em>Molecular Ecology</em> 18 (19): 4061–72. <a href="https://doi.org/10.1111/j.1365-294X.2009.04311.x" class="external-link">https://doi.org/10.1111/j.1365-294X.2009.04311.x</a>.
</div>
<div id="ref-TerBraak2004" class="csl-entry">
Ter Braak, Cajo J. F., and I.Colin Prentice. 2004. <span>“A Theory of
Gradient Analysis.”</span> In <em>Advances in Ecological Research:
Classic Papers</em>, 34:235–82. Advances in Ecological Research.
Academic Press. <a href="https://doi.org/10.1016/S0065-2504(03)34003-6" class="external-link">https://doi.org/10.1016/S0065-2504(03)34003-6</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://darulab.org/" class="external-link">Barnabas H. Daru</a>, <a href="https://github.com/piyalkarum/" class="external-link">Piyal Karunarathne</a>, <a href="https://kschliep.netlify.app/" class="external-link">Klaus Schliep</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
