<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction to the phyloregion package â€¢ phyloregion</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to the phyloregion package">
<meta property="og:description" content="phyloregion">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phyloregion</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Benchmark.html">Benchmarking phyloregion</a>
    </li>
    <li>
      <a href="../articles/heterogeneity.html">Endemism patterns are scale dependent</a>
    </li>
    <li>
      <a href="../articles/phyloregion-intro.html">An introduction to the phyloregion package</a>
    </li>
    <li>
      <a href="../articles/scalability.html">Scalability of phyloregion</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/darunabas/phyloregion/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="phyloregion-intro_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An introduction to the phyloregion package</h1>
                        <h4 class="author">Barnabas H. Daru, Piyal Karunarathne &amp; Klaus Schliep</h4>
            
            <h4 class="date">July 19, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/darunabas/phyloregion/blob/master/vignettes/phyloregion-intro.Rmd"><code>vignettes/phyloregion-intro.Rmd</code></a></small>
      <div class="hidden name"><code>phyloregion-intro.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>1. Installation</h2>
<p><code>phyloregion</code> is available from the <a href="https://CRAN.R-project.org/package=phyloregion">Comprehensive R Archive Network</a>, so you can use the following line of code to install and run it:</p>
<pre><code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("phyloregion")</a></code></pre>
<p>Alternatively, you can install the development version of <code>phyloregion</code> hosted on GitHub. To do this, you will need to install the <code>devtools</code> package. In R, type:</p>
<pre><code>if (!requireNamespace("remotes", quietly = TRUE)) 
    install.packages("remotes") 
remotes::install_github("darunabas/phyloregion")</code></pre>
<p>When installed, load the package in R:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">phyloregion</span>)</pre></body></html></div>
</div>
<div id="overview-and-general-workflow-of-phyloregion" class="section level2">
<h2 class="hasAnchor">
<a href="#overview-and-general-workflow-of-phyloregion" class="anchor"></a>2. Overview and general workflow of <code>phyloregion</code>
</h2>
<p>The workflow of the <code>phyloregion</code> package demonstrates steps from preparation of different types of data to visualizing the results of biogeographical regionalization, together with tips on selecting the optimal method for achieving the best output, depending on the types of data used and research questions.</p>
<div class="figure">
<img src="workflow.png" alt=""><p class="caption"><strong>Figure 1.</strong> Simplified workflow for analysis of biogeographical regionalization using phyloregion. Distribution data is converted to a sparse community matrix. When paired with phylogenetic data, phylobuilder creates a subtree with largest overlap from a species list, thereby ensuring complete representation of missing data; phylocommunity matrix to visualization of results.</p>
</div>
</div>
<div id="input-data" class="section level2">
<h2 class="hasAnchor">
<a href="#input-data" class="anchor"></a>3. Input data</h2>
<div id="phylogenies" class="section level3">
<h3 class="hasAnchor">
<a href="#phylogenies" class="anchor"></a>Phylogenies</h3>
<p>In R, phylogenetic relationships among species / taxa are often represented as a phylo object implemented in the <code>ape</code> package<span class="citation"><sup>1</sup></span>. Phylogenies (often in the Newick or Nexus formats) can be imported into R with the <code>read.tree</code> or <code>read.nexus</code> functions of the <code>ape</code> package<span class="citation"><sup>1</sup></span>.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ape</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Matrix</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">sp</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">africa</span>)
<span class="no">sparse_comm</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">comm</span>

<span class="no">tree</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">phylo</span>
<span class="no">tree</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/drop.tip.html">keep.tip</a></span>(<span class="no">tree</span>, <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span>(<span class="no">tree</span>$<span class="no">tip.label</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">sparse_comm</span>)))
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>))
<span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">tree</span>, <span class="kw">show.tip.label</span><span class="kw">=</span><span class="fl">FALSE</span>)</pre></body></html></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-2-1.png" alt="__Figure 2.__ Phylogenetic tree of the woody plants of southern Africa inferred from DNA barcodes using a maximum likelihood approach and transforming branch lengths to millions of years ago by enforcing a relaxed molecular clock and multiple calibrations [@Daru2015ddi]." width="700"><p class="caption">
<strong>Figure 2.</strong> Phylogenetic tree of the woody plants of southern Africa inferred from DNA barcodes using a maximum likelihood approach and transforming branch lengths to millions of years ago by enforcing a relaxed molecular clock and multiple calibrations.<span class="citation"><sup>2</sup></span>
</p>
</div>
</div>
<div id="distribution-data-input" class="section level3">
<h3 class="hasAnchor">
<a href="#distribution-data-input" class="anchor"></a>Distribution data input</h3>
<p>The <code>phyloregion</code> package has functions for manipulating three kinds of distribution data: point records, polygons and raster layers. An overview can be easily obtained with the functions <code>points2comm</code>, <code>polys2comm</code> and <code>raster2comm</code> for point records, polygons, or raster layers, respectively. Depending on the data source, all three functions ultimately provide convenient interfaces to convert the distribution data to a community matrix at varying spatial grains and extents for downstream analyses.</p>
<p>We will play around with these functions in turn.</p>
</div>
<div id="function-points2comm" class="section level3">
<h3 class="hasAnchor">
<a href="#function-points2comm" class="anchor"></a><strong><em>Function <code>points2comm</code></em></strong>
</h3>
<p>Here, we will generate random points in geographic space, similar to data obtained from museum records, GBIF, iDigBio, or CIESIN which typically have columns of geographic coordinates for each observation.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">s</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"nigeria/nigeria.rds"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"phyloregion"</span>))

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">1</span>)
<span class="no">m</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw pkg">sp</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/sp/man/spsample.html">spsample</a></span>(<span class="no">s</span>, <span class="fl">10000</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"nonaligned"</span>))
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">m</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"lon"</span>, <span class="st">"lat"</span>)
<span class="no">species</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"sp"</span>, <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="fl">1</span>:<span class="fl">1000</span>))
<span class="no">m</span>$<span class="no">taxon</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="no">species</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">m</span>), <span class="kw">replace</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="no">pt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/raster2comm.html">points2comm</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">m</span>, <span class="kw">mask</span> <span class="kw">=</span> <span class="no">s</span>, <span class="kw">res</span> <span class="kw">=</span> <span class="fl">0.5</span>, <span class="kw">lon</span> <span class="kw">=</span> <span class="st">"lon"</span>, <span class="kw">lat</span> <span class="kw">=</span> <span class="st">"lat"</span>,
            <span class="kw">species</span> <span class="kw">=</span> <span class="st">"taxon"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">pt</span><span class="kw">[[</span><span class="fl">1</span>]][<span class="fl">1</span>:<span class="fl">5</span>, <span class="fl">1</span>:<span class="fl">5</span>])</pre></body></html></div>
<pre><code>## 5 x 5 sparse Matrix of class "dgCMatrix"
##      sp1 sp10 sp100 sp1000 sp101
## v100   .    .     .      .     1
## v101   .    .     .      .     .
## v102   .    .     .      .     .
## v103   .    1     .      .     .
## v104   .    .     .      .     .</code></pre>
</div>
<div id="function-polys2comm" class="section level3">
<h3 class="hasAnchor">
<a href="#function-polys2comm" class="anchor"></a><strong><em>Function <code>polys2comm</code></em></strong>
</h3>
<p>This function converts polygons to a community matrix at varying spatial grains and extents for downstream analyses. Polygons can be derived from the IUCN Redlist spatial database (https: //www.iucnredlist.org/resources/spatial-data-download), published monographs or field guides validated by taxonomic experts. To illustrate this function, we will use the function <code>random_species</code> to generate random polygons for five random species over the landscape of Nigeria as follows:</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">s</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"nigeria/nigeria.rds"</span>, <span class="kw">package</span><span class="kw">=</span><span class="st">"phyloregion"</span>))
<span class="no">sp</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/random_species.html">random_species</a></span>(<span class="fl">100</span>, <span class="kw">species</span><span class="kw">=</span><span class="fl">5</span>, <span class="kw">shp</span><span class="kw">=</span><span class="no">s</span>)
<span class="no">pol</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/raster2comm.html">polys2comm</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">sp</span>, <span class="kw">species</span> <span class="kw">=</span> <span class="st">"species"</span>, <span class="kw">trace</span><span class="kw">=</span><span class="fl">0</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">pol</span><span class="kw">[[</span><span class="fl">1</span>]][<span class="fl">1</span>:<span class="fl">5</span>, <span class="fl">1</span>:<span class="fl">5</span>])</pre></body></html></div>
<pre><code>## 5 x 5 sparse Matrix of class "dgCMatrix"
##     species1 species2 species3 species4 species5
## v10        .        .        1        .        .
## v12        1        .        1        1        1
## v13        1        .        1        1        1
## v14        1        .        1        1        1
## v15        1        .        1        1        1</code></pre>
</div>
<div id="function-raster2comm" class="section level3">
<h3 class="hasAnchor">
<a href="#function-raster2comm" class="anchor"></a><strong><em>Function <code>raster2comm</code></em></strong>
</h3>
<p>This third function, converts raster layers (often derived from species distribution modeling, such as aquamaps<span class="citation"><sup>3</sup></span>) to a community matrix.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">fdir</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"NGAplants"</span>, <span class="kw">package</span><span class="kw">=</span><span class="st">"phyloregion"</span>)
<span class="no">files</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="no">fdir</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>(<span class="no">fdir</span>))
<span class="no">ras</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/raster2comm.html">raster2comm</a></span>(<span class="no">files</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">ras</span><span class="kw">[[</span><span class="fl">1</span>]])</pre></body></html></div>
<pre><code>## 6 x 16 sparse Matrix of class "dgCMatrix"</code></pre>
<pre><code>##    [[ suppressing 16 column names 'Chytranthus_gilletii', 'Commelina_ramulosa', 'Cymbopogon_caesius' ... ]]</code></pre>
<pre><code>##                                     
## v100 . . . 1 . . . . . . . . . . . .
## v101 . . . 1 . . . . . . . . . . . .
## v102 . . . 1 . . . . . . . . . . . .
## v103 . . . 1 . . . . . . . . . . . .
## v104 . . . 1 . . . . . . . . . . . .
## v105 . . . 1 . . . . . . . . . . . .</code></pre>
<p>The object <code>ras</code> above also returns two objects: a community data frame and a shapefile of grid cells with the numbers of species per cell and can be plotted as a heatmap using our nice <code>plot_swatch</code> function as follows:</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="no">s</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"nigeria/SR_Naija.rds"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"phyloregion"</span>))
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">0</span>,<span class="fl">4</span>))
<span class="fu"><a href="../reference/plot_swatch.html">plot_swatch</a></span>(<span class="no">s</span>, <span class="kw">values</span> <span class="kw">=</span> <span class="no">s</span>$<span class="no">SR</span>, <span class="kw">k</span> <span class="kw">=</span> <span class="fl">20</span>, <span class="kw">leg</span><span class="kw">=</span><span class="fl">1</span>, <span class="kw">border</span><span class="kw">=</span><span class="fl">NA</span>)</pre></body></html></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-6-1.png" alt="__Figure 3.__ Species richness of plants in Nigeria across equal area grid cells. This is to demonstrate how the function `plot_swatch` works." width="50%"><p class="caption">
<strong>Figure 3.</strong> Species richness of plants in Nigeria across equal area grid cells. This is to demonstrate how the function <code>plot_swatch</code> works.
</p>
</div>
</div>
<div id="community-data" class="section level3">
<h3 class="hasAnchor">
<a href="#community-data" class="anchor"></a>Community data</h3>
<p>Community data are commonly stored in a matrix with the sites as rows and species / operational taxonomic units (OTUs) as columns. The elements of the matrix are numeric values indicating the abundance/observations or presence/absence (0/1) of OTUs in different sites. In practice, such a matrix can contain many zero values because species are known to generally have unimodal distributions along environmental gradients,<span class="citation"><sup>4</sup></span> and storing and analyzing every single element of that matrix can be computationally challenging and expensive.</p>
<p><code>phyloregion</code> differs from other R packages (e.g.Â vegan,<span class="citation"><sup>5</sup></span> picante<span class="citation"><sup>6</sup></span> or betapart<span class="citation"><sup>7</sup></span>) in that the data are not stored in a (dense) <code>matrix</code> or <code>data.frame</code> but as a sparse matrix making use of the infrastructure provided by the Matrix package.<span class="citation"><sup>8</sup></span> A sparse matrix is a matrix with a high proportion of zero entries<span class="citation"><sup>9</sup></span>, of which only the non-zero entries are stored and used for downstream analysis.</p>
<p>A sparse matrix representation has two advantages. First the community matrix can be stored in a much memory efficient manner, allowing analysis of larger datasets. Second, for very large datasets spanning thousands of taxa and spatial scales, computations with a sparse matrix are often much faster.<br>
The <code>phyloregion</code> package contains functions to conveniently change between data formats.</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Matrix</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">africa</span>)
<span class="no">sparse_comm</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">comm</span>
<span class="no">dense_comm</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="no">sparse_comm</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/object.size.html">object.size</a></span>(<span class="no">dense_comm</span>)</pre></body></html></div>
<pre><code>## 4216952 bytes</code></pre>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/object.size.html">object.size</a></span>(<span class="no">sparse_comm</span>)</pre></body></html></div>
<pre><code>## 885952 bytes</code></pre>
<p>Here, the data set in the dense matrix representation consumes roughly five times more memory than the sparse representation.</p>
</div>
</div>
<div id="analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis" class="anchor"></a>4. Analysis</h2>
<div id="alpha-diversity" class="section level3">
<h3 class="hasAnchor">
<a href="#alpha-diversity" class="anchor"></a>Alpha diversity</h3>
<p>We demonstrate the utility of <code>phyloregion</code> in mapping standard conservation metrics of species richness, weighted endemism (<code>weighted_endemism</code>) and threat (<code>map_traits</code>) as well as fast computations of phylodiversity measures such as phylogenetic diversity (<code>PD</code>), phylogenetic endemism (<code>phylo_endemism</code>), and evolutionary distinctiveness and global endangerment (<code>EDGE</code>). The major advantage of these functions compared to available tools e.g.Â biodiverse,<span class="citation"><sup>10</sup></span> is the ability to utilize sparse matrix that speeds up the analyses without exhausting computer memories, making it ideal for handling any data from small local scales to large regional and global scales.</p>
<div id="function-weighted_endemism" class="section level4">
<h4 class="hasAnchor">
<a href="#function-weighted_endemism" class="anchor"></a>Function <code>weighted_endemism</code>
</h4>
<p>Weighted endemism is species richness inversely weighted by species ranges<span class="citation"><sup>11</sup></span>,<span class="citation"><sup>12</sup></span>,<span class="citation"><sup>13</sup></span>.</p>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">raster</span>)</pre></body></html></div>
<pre><code>## 
## Attaching package: 'raster'</code></pre>
<pre><code>## The following objects are masked from 'package:ape':
## 
##     rotate, zoom</code></pre>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">africa</span>)
<span class="no">Endm</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/weighted_endemism.html">weighted_endemism</a></span>(<span class="no">africa</span>$<span class="no">comm</span>)
<span class="fu"><a href="https://rdrr.io/pkg/raster/man/headtail.html">head</a></span>(<span class="no">Endm</span>)</pre></body></html></div>
<pre><code>##    v3635    v3636    v3637    v3638    v3639    v3640 
## 1.770041 2.637894 1.825862 1.270093 1.043782 0.259324</code></pre>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="no">m</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/raster/man/merge.html">merge</a></span>(<span class="no">africa</span>$<span class="no">polys</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">grids</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/raster/man/names.html">names</a></span>(<span class="no">Endm</span>), <span class="kw">WE</span><span class="kw">=</span><span class="no">Endm</span>), <span class="kw">by</span><span class="kw">=</span><span class="st">"grids"</span>)
<span class="no">m</span> <span class="kw">&lt;-</span> <span class="no">m</span>[!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">m</span>@<span class="kw">data</span>$<span class="no">WE</span>),]

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">0</span>,<span class="fl">4</span>))
<span class="fu"><a href="../reference/plot_swatch.html">plot_swatch</a></span>(<span class="no">m</span>, <span class="kw">values</span> <span class="kw">=</span> <span class="no">m</span>$<span class="no">WE</span>, <span class="kw">k</span><span class="kw">=</span><span class="fl">20</span>, <span class="kw">leg</span> <span class="kw">=</span> <span class="fl">3</span>, <span class="kw">border</span> <span class="kw">=</span> <span class="fl">NA</span>)</pre></body></html></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-8-1.png" alt="__Figure 4.__ Geographic distributions of weighted endemism for woody plants of southern Africa." width="50%"><p class="caption">
<strong>Figure 4.</strong> Geographic distributions of weighted endemism for woody plants of southern Africa.
</p>
</div>
</div>
<div id="function-pd-phylogenetic-diversity" class="section level4">
<h4 class="hasAnchor">
<a href="#function-pd-phylogenetic-diversity" class="anchor"></a>Function <code>PD</code> â€“ phylogenetic diversity</h4>
<p>Phylogenetic diversity (<code>PD</code>) represents the length of evolutionary pathways that connects a given set of taxa on a rooted phylogenetic tree.<span class="citation"><sup>14</sup></span> This metric is often characterised in units of time (millions of years, for dated phylogenies). We will map PD for plants of southern Africa.</p>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">africa</span>)
<span class="no">comm</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">comm</span>
<span class="no">tree</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">phylo</span>
<span class="no">poly</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">polys</span>

<span class="no">mypd</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/PD.html">PD</a></span>(<span class="no">comm</span>, <span class="no">tree</span>)
<span class="fu"><a href="https://rdrr.io/pkg/raster/man/headtail.html">head</a></span>(<span class="no">mypd</span>)</pre></body></html></div>
<pre><code>##    v3635    v3636    v3637    v3638    v3639    v3640 
## 4226.216 5372.009 4377.735 3783.992 3260.111 1032.685</code></pre>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="no">M</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/raster/man/merge.html">merge</a></span>(<span class="no">poly</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">grids</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/raster/man/names.html">names</a></span>(<span class="no">mypd</span>), <span class="kw">pd</span><span class="kw">=</span><span class="no">mypd</span>), <span class="kw">by</span><span class="kw">=</span><span class="st">"grids"</span>)
<span class="no">M</span> <span class="kw">&lt;-</span> <span class="no">M</span>[!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">M</span>@<span class="kw">data</span>$<span class="no">pd</span>),]
<span class="fu"><a href="https://rdrr.io/pkg/raster/man/headtail.html">head</a></span>(<span class="no">M</span>)</pre></body></html></div>
<pre><code>##   grids       pd
## 1 v3635 4226.216
## 2 v3636 5372.009
## 3 v3637 4377.735
## 4 v3638 3783.992
## 5 v3639 3260.111
## 6 v3640 1032.685</code></pre>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">0</span>,<span class="fl">4</span>))
<span class="fu"><a href="../reference/plot_swatch.html">plot_swatch</a></span>(<span class="no">M</span>, <span class="kw">values</span> <span class="kw">=</span> <span class="no">M</span>$<span class="no">pd</span>, <span class="kw">k</span><span class="kw">=</span><span class="fl">20</span>, <span class="kw">border</span><span class="kw">=</span><span class="fl">NA</span>, <span class="kw">leg</span><span class="kw">=</span><span class="fl">3</span>)</pre></body></html></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-9-1.png" alt="__Figure 5.__ Geographic distributions of phylogenetic diversity for woody plants of southern Africa." width="50%"><p class="caption">
<strong>Figure 5.</strong> Geographic distributions of phylogenetic diversity for woody plants of southern Africa.
</p>
</div>
</div>
<div id="function-phylo_endemism-phylogenetic-endemism" class="section level4">
<h4 class="hasAnchor">
<a href="#function-phylo_endemism-phylogenetic-endemism" class="anchor"></a>Function <code>phylo_endemism</code> â€“ phylogenetic endemism</h4>
<p>Phylogenetic endemism is not influenced by variations in taxonomic opinion because it measures endemism based on the relatedness of species before weighting it by their range sizes<span class="citation"><sup>15</sup></span>,<span class="citation"><sup>13</sup></span>.</p>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">raster</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">africa</span>)
<span class="no">comm</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">comm</span>
<span class="no">tree</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">phylo</span>
<span class="no">poly</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">polys</span>

<span class="no">pe</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/phylo_endemism.html">phylo_endemism</a></span>(<span class="no">comm</span>, <span class="no">tree</span>)
<span class="fu"><a href="https://rdrr.io/pkg/raster/man/headtail.html">head</a></span>(<span class="no">pe</span>)</pre></body></html></div>
<pre><code>##     v3635     v3636     v3637     v3638     v3639     v3640 
## 32.536530 45.262625 35.004944 27.603721 23.183947  6.439589</code></pre>
<div class="sourceCode" id="cb31"><html><body><pre class="r"><span class="no">mx</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/raster/man/merge.html">merge</a></span>(<span class="no">poly</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">grids</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/raster/man/names.html">names</a></span>(<span class="no">pe</span>), <span class="kw">pe</span><span class="kw">=</span><span class="no">pe</span>), <span class="kw">by</span><span class="kw">=</span><span class="st">"grids"</span>)
<span class="no">mx</span> <span class="kw">&lt;-</span> <span class="no">mx</span>[!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">mx</span>@<span class="kw">data</span>$<span class="no">pe</span>),]
<span class="fu"><a href="https://rdrr.io/pkg/raster/man/headtail.html">head</a></span>(<span class="no">mx</span>)</pre></body></html></div>
<pre><code>##   grids        pe
## 1 v3635 32.536530
## 2 v3636 45.262625
## 3 v3637 35.004944
## 4 v3638 27.603721
## 5 v3639 23.183947
## 6 v3640  6.439589</code></pre>
<div class="sourceCode" id="cb33"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">0</span>,<span class="fl">4</span>))
<span class="fu"><a href="../reference/plot_swatch.html">plot_swatch</a></span>(<span class="no">mx</span>, <span class="kw">values</span> <span class="kw">=</span> <span class="no">mx</span>$<span class="no">pe</span>, <span class="kw">k</span><span class="kw">=</span><span class="fl">20</span>, <span class="kw">border</span><span class="kw">=</span><span class="fl">NA</span>, <span class="kw">leg</span><span class="kw">=</span><span class="fl">3</span>)</pre></body></html></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-10-1.png" alt="__Figure 6.__ Geographic distributions of phylogenetic endemism for woody plants of southern Africa." width="50%"><p class="caption">
<strong>Figure 6.</strong> Geographic distributions of phylogenetic endemism for woody plants of southern Africa.
</p>
</div>
</div>
<div id="function-edge-evolutionary-distinctiveness-and-global-endangerment" class="section level4">
<h4 class="hasAnchor">
<a href="#function-edge-evolutionary-distinctiveness-and-global-endangerment" class="anchor"></a>Function <code>EDGE</code> â€“ Evolutionary Distinctiveness and Global Endangerment</h4>
<p>This function calculates EDGE by combining evolutionary distinctiveness (ED; i.e., phylogenetic isolation of a species) with global endangerment (GE) status as defined by the International Union for Conservation of Nature (IUCN).</p>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">africa</span>)
<span class="no">comm</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">comm</span>
<span class="no">threat</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">IUCN</span>
<span class="no">tree</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">phylo</span>
<span class="no">poly</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">polys</span>

<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/EDGE.html">EDGE</a></span>(<span class="no">threat</span>, <span class="no">tree</span>, <span class="kw">Redlist</span> <span class="kw">=</span> <span class="st">"IUCN"</span>, <span class="kw">species</span><span class="kw">=</span><span class="st">"Species"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/raster/man/headtail.html">head</a></span>(<span class="no">x</span>)</pre></body></html></div>
<pre><code>##        Abutilon_angulatum_OM1934    Abutilon_sonneratianum_LTM034 
##                         2.903551                         2.903551 
## Acalypha_glabrata_glabrata_OM441  Acalypha_glabrata_pilosa_OM1979 
##                         2.480505                         2.480505 
##       Acalypha_sonderiana_OM2163  Acokanthera_oblongifolia_OM2240 
##                         2.914481                         2.211561</code></pre>
<div class="sourceCode" id="cb36"><html><body><pre class="r"><span class="no">y</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/map_trait.html">map_trait</a></span>(<span class="no">comm</span>, <span class="no">x</span>, <span class="kw">FUN</span> <span class="kw">=</span> <span class="no">sd</span>, <span class="kw">shp</span><span class="kw">=</span><span class="no">poly</span>)

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">0</span>,<span class="fl">4</span>))
<span class="fu"><a href="../reference/plot_swatch.html">plot_swatch</a></span>(<span class="no">y</span>, <span class="no">y</span>$<span class="no">traits</span>, <span class="kw">k</span><span class="kw">=</span><span class="fl">20</span>, <span class="kw">border</span><span class="kw">=</span><span class="fl">NA</span>, <span class="kw">leg</span><span class="kw">=</span><span class="fl">3</span>)</pre></body></html></div>
<div class="figure" style="text-align: left">
<img src="phyloregion-intro_files/figure-html/unnamed-chunk-11-1.png" alt="__Figure 7.__ Geographic distributions of evolutionary distinctiveness and global endangerment for woody plants of southern Africa." width="50%"><p class="caption">
<strong>Figure 7.</strong> Geographic distributions of evolutionary distinctiveness and global endangerment for woody plants of southern Africa.
</p>
</div>
</div>
</div>
<div id="analysis-of-beta-diversity-phylogenetic-and-non-phylogenetic" class="section level3">
<h3 class="hasAnchor">
<a href="#analysis-of-beta-diversity-phylogenetic-and-non-phylogenetic" class="anchor"></a>Analysis of beta diversity (phylogenetic and non-phylogenetic)</h3>
<p>The three commonly used methods for quantifying -diversity, the variation in species composition among sites, â€“ Simpson, Sorenson and Jaccard<span class="citation"><sup>16</sup></span>. The <code>phyloregion</code>â€™s functions <code>beta_diss</code> and <code>phylobeta</code> compute efficiently pairwise dissimilarities matrices for large sparse community matrices and phylogenetic trees for taxonomic and phylogenetic turnover, respectively. The results are stored as distance objects for subsequent analyses.</p>
<hr>
</div>
<div id="phylogenetic-beta-diversity" class="section level3">
<h3 class="hasAnchor">
<a href="#phylogenetic-beta-diversity" class="anchor"></a>Phylogenetic beta diversity</h3>
<p><code>phyloregion</code> offers a fast means of computing phylogenetic beta diversity, the turnover of branch lengths among sites, making use of and improving on the infrastructure provided by the <code>betapart</code> package<span class="citation"><sup>7</sup></span> allowing a sparse community matrix as input.</p>
<div class="sourceCode" id="cb37"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">africa</span>)
<span class="no">sparse_comm</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">comm</span>

<span class="no">tree</span> <span class="kw">&lt;-</span> <span class="no">africa</span>$<span class="no">phylo</span>
<span class="no">tree</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/drop.tip.html">keep.tip</a></span>(<span class="no">tree</span>, <span class="fu"><a href="https://rdrr.io/pkg/raster/man/intersect.html">intersect</a></span>(<span class="no">tree</span>$<span class="no">tip.label</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">sparse_comm</span>)))
<span class="no">pb</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/phylobeta.html">phylobeta</a></span>(<span class="no">sparse_comm</span>, <span class="no">tree</span>)</pre></body></html></div>
<div class="sourceCode" id="cb38"><html><body><pre class="r"><span class="no">y</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/phyloregion.html">phyloregion</a></span>(<span class="no">pb</span><span class="kw">[[</span><span class="fl">1</span>]], <span class="kw">shp</span><span class="kw">=</span><span class="no">africa</span>$<span class="no">polys</span>)</pre></body></html></div>
<div class="sourceCode" id="cb39"><html><body><pre class="r"><span class="fu"><a href="../reference/plot.phyloregion.html">plot_NMDS</a></span>(<span class="no">y</span>, <span class="kw">cex</span><span class="kw">=</span><span class="fl">3</span>)
<span class="fu"><a href="../reference/plot.phyloregion.html">text_NMDS</a></span>(<span class="no">y</span>)</pre></body></html></div>
<p><img src="phyloregion-intro_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<div class="sourceCode" id="cb40"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">0</span>,<span class="fl">4</span>))
<span class="fu"><a href="https://rdrr.io/pkg/raster/man/plot.html">plot</a></span>(<span class="no">y</span>, <span class="kw">palette</span><span class="kw">=</span><span class="st">"NMDS"</span>)</pre></body></html></div>
<p><img src="phyloregion-intro_files/figure-html/unnamed-chunk-14-2.png" width="700"></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h2>
<div class="sourceCode" id="cb41"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</pre></body></html></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] raster_3.3-7      sp_1.4-2          Matrix_1.2-18     ape_5.4          
## [5] phyloregion_1.0.4
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5       highr_0.8        compiler_4.0.2   tools_4.0.2     
##  [5] magic_1.5-9      digest_0.6.25    betapart_1.5.1   evaluate_0.14   
##  [9] memoise_1.1.0    nlme_3.1-148     lattice_0.20-41  mgcv_1.8-31     
## [13] pkgconfig_2.0.3  rlang_0.4.6      igraph_1.2.5     fastmatch_1.1-0 
## [17] rstudioapi_0.11  rgdal_1.5-12     yaml_2.2.1       parallel_4.0.2  
## [21] pkgdown_1.5.1    xfun_0.15        stringr_1.4.0    knitr_1.29      
## [25] cluster_2.1.0    rgeos_0.5-3      desc_1.2.0       fs_1.4.2        
## [29] rcdd_1.2-2       rprojroot_1.3-2  grid_4.0.2       R6_2.4.1        
## [33] phangorn_2.5.5   rmarkdown_2.3    magrittr_1.5     codetools_0.2-16
## [37] backports_1.1.8  htmltools_0.5.0  MASS_7.3-51.6    splines_4.0.2   
## [41] assertthat_0.2.1 abind_1.4-5      colorspace_1.4-1 picante_1.8.2   
## [45] permute_0.9-5    quadprog_1.5-8   stringi_1.4.6    geometry_0.4.5  
## [49] crayon_1.3.4     vegan_2.5-6</code></pre>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>REFERENCES</h2>
<div id="refs" class="references">
<div id="ref-Paradis2018">
<p>1. Paradis, E. &amp; Schliep, K. Ape 5.0: An environment for modern phylogenetics and evolutionary analyses in R. <em>Bioinformatics</em> <strong>35</strong>, 526â€“528 (2018).</p>
</div>
<div id="ref-Daru2015ddi">
<p>2. Daru, B. H., Bank, M. van der &amp; Davies, T. J. Spatial incongruence among hotspots and complementary areas of tree diversity in southern africa. <em>Diversity and Distributions</em> <strong>21</strong>, 769â€“780 (2015).</p>
</div>
<div id="ref-kaschner2008aquamaps">
<p>3. Kaschner, K. <em>et al.</em> AquaMaps: Predicted range maps for aquatic species. <em>World wide web electronic publication, wwwaquamapsorg, Version</em> <strong>10</strong>, 2008 (2008).</p>
</div>
<div id="ref-TerBraak2004">
<p>4. Ter Braak, C. J. F. &amp; Prentice, I. A theory of gradient analysis. in <em>Advances in ecological research: Classic papers</em> vol. 34 235â€“282 (Academic Press, 2004).</p>
</div>
<div id="ref-vegan">
<p>5. Oksanen, J. <em>et al.</em> <em>Vegan: Community ecology package</em>. (2019).</p>
</div>
<div id="ref-Kembel2010">
<p>6. Kembel, S. W. <em>et al.</em> Picante: R tools for integrating phylogenies and ecology. <em>Bioinformatics</em> <strong>26</strong>, 1463â€“1464 (2010).</p>
</div>
<div id="ref-Baselga2012">
<p>7. Baselga, A. &amp; Orme, C. D. L. Betapart: An r package for the study of beta diversity. <em>Methods in Ecology and Evolution</em> <strong>3</strong>, 808â€“812 (2012).</p>
</div>
<div id="ref-Matrix">
<p>8. Bates, D. &amp; Maechler, M. <em>Matrix: Sparse and dense matrix classes and methods</em>. (2019).</p>
</div>
<div id="ref-Duff1977">
<p>9. Duff, I. S. A survey of sparse matrix research. <em>Proceedings of the IEEE</em> <strong>65</strong>, 500â€“535 (1977).</p>
</div>
<div id="ref-laffan2010biodiverse">
<p>10. Laffan, S. W., Lubarsky, E. &amp; Rosauer, D. F. Biodiverse, a tool for the spatial analysis of biological and related diversity. <em>Ecography</em> <strong>33</strong>, 643â€“647 (2010).</p>
</div>
<div id="ref-crisp2001endemism">
<p>11. Crisp, M. D., Laffan, S., Linder, H. P. &amp; Monro, A. Endemism in the australian flora. <em>Journal of Biogeography</em> <strong>28</strong>, 183â€“198 (2001).</p>
</div>
<div id="ref-laffan2003assessing">
<p>12. Laffan, S. W. &amp; Crisp, M. D. Assessing endemism at multiple spatial scales, with an example from the australian vascular flora. <em>Journal of Biogeography</em> <strong>30</strong>, 511â€“520 (2003).</p>
</div>
<div id="ref-daru2020endemism">
<p>13. Daru, B. H., Farooq, H., Antonelli, A. &amp; Faurby, S. Endemism patterns are scale dependent. <em>Nature Communications</em> <strong>11</strong>, 2115 (2020).</p>
</div>
<div id="ref-Faith1992">
<p>14. Faith, D. P. Conservation evaluation and phylogenetic diversity. <em>Biological Conservation</em> <strong>61</strong>, 1â€“10 (1992).</p>
</div>
<div id="ref-Rosauer2009">
<p>15. Rosauer, D., Laffan, S. W., Crisp, M. D., Donnellan, S. C. &amp; Cook, L. G. Phylogenetic endemism: A new approach for identifying geographical concentrations of evolutionary history. <em>Molecular Ecology</em> <strong>18</strong>, 4061â€“4072 (2009).</p>
</div>
<div id="ref-laffan2016range">
<p>16. Laffan, S. W. <em>et al.</em> Range-weighted metrics of species and phylogenetic turnover can better resolve biogeographic transition zones. <em>Methods in Ecology and Evolution</em> <strong>7</strong>, 580â€“588 (2016).</p>
</div>
<div id="ref-Daru2020">
<p>17. Daru, B. H., Karunarathne, P. &amp; Schliep, K. Phyloregion: R package for biogeographic regionalization and macroecology. <em>bioRxiv</em> (2020) doi:<a href="https://doi.org/10.1101/2020.02.12.945691">10.1101/2020.02.12.945691</a>.</p>
</div>
<div id="ref-Daru2017">
<p>18. Daru, B. H., Elliott, T. L., Park, D. S. &amp; Davies, T. J. Understanding the processes underpinning patterns of phylogenetic regionalization. <em>Trends in ecology &amp; evolution</em> <strong>32</strong>, 845â€“860 (2017).</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Barnabas H. Daru, Piyal Karunarathne, Klaus Schliep.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
